id;post_id;username;body;created_at;updated_at
510116274;510116274;dependabot[bot];<p>Bumps <a href="https://github.com/urllib3/urllib3">urllib3</a> from 1.23 to 1.24.2.
<details>
<summary>Changelog</summary></p>
<p><em>Sourced from <a href="https://github.com/urllib3/urllib3/blob/master/CHANGES.rst">urllib3's changelog</a>.</em></p>
<blockquote>
<h2>1.24.2 (2019-04-17)</h2>
<ul>
<li>
<p>Don't load system certificates by default when any other <code>ca_certs</code>, <code>ca_certs_dir</code> or
  <code>ssl_context</code> parameters are specified.</p>
</li>
<li>
<p>Remove Authorization header regardless of case when redirecting to cross-site. (Issue <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1510">#1510</a>)</p>
</li>
<li>
<p>Add support for IPv6 addresses in subjectAltName section of certificates. (Issue <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1269">#1269</a>)</p>
</li>
</ul>
<h2>1.24.1 (2018-11-02)</h2>
<ul>
<li>
<p>Remove quadratic behavior within <code>GzipDecoder.decompress()</code> (Issue <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1467">#1467</a>)</p>
</li>
<li>
<p>Restored functionality of <code>ciphers</code> parameter for <code>create_urllib3_context()</code>. (Issue <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1462">#1462</a>)</p>
</li>
</ul>
<h2>1.24 (2018-10-16)</h2>
<ul>
<li>
<p>Allow key_server_hostname to be specified when initializing a PoolManager to allow custom SNI to be overridden. (Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1449">#1449</a>)</p>
</li>
<li>
<p>Test against Python 3.7 on AppVeyor. (Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1453">#1453</a>)</p>
</li>
<li>
<p>Early-out ipv6 checks when running on App Engine. (Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1450">#1450</a>)</p>
</li>
<li>
<p>Change ambiguous description of backoff_factor (Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1436">#1436</a>)</p>
</li>
<li>
<p>Add ability to handle multiple Content-Encodings (Issue <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1441">#1441</a> and Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1442">#1442</a>)</p>
</li>
<li>
<p>Skip DNS names that can't be idna-decoded when using pyOpenSSL (Issue <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1405">#1405</a>).</p>
</li>
<li>
<p>Add a server_hostname parameter to HTTPSConnection which allows for
  overriding the SNI hostname sent in the handshake. (Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1397">#1397</a>)</p>
</li>
<li>
<p>Drop support for EOL Python 2.6 (Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1429">#1429</a> and Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1430">#1430</a>)</p>
</li>
<li>
<p>Fixed bug where responses with header Content-Type: message/* erroneously
  raised HeaderParsingError, resulting in a warning being logged. (Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1439">#1439</a>)</p>
</li>
<li>
<p>Move urllib3 to src/urllib3 (Pull <a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1409">#1409</a>)
</details>
<details>
<summary>Commits</summary></p>
</li>
</ul>
</blockquote>
<ul>
<li><a href="https://github.com/urllib3/urllib3/commit/1efadf43dc63317cd9eaa3e0fdb9e05ab07254b1"><code>1efadf4</code></a> Release 1.24.2 (<a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1564">#1564</a>)</li>
<li><a href="https://github.com/urllib3/urllib3/commit/a6ec68a5c5c5743c59fe5c62c635c929586c429b"><code>a6ec68a</code></a> Merging new release version: 1.24.1</li>
<li><a href="https://github.com/urllib3/urllib3/commit/0cedb3b0f1e5d79c89c6db767c534b064b794cf2"><code>0cedb3b</code></a> Restore context.set_ciphers() to create_urllib3_context() (<a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1463">#1463</a>)</li>
<li><a href="https://github.com/urllib3/urllib3/commit/0aeba3be0224a930f6ffef254ed12b41303a86d7"><code>0aeba3b</code></a> Use bytearray to accumulate bytes from gzip (<a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1468">#1468</a>)</li>
<li><a href="https://github.com/urllib3/urllib3/commit/f8d1c787d9b02a70d66ddbde9c99061d9073d54a"><code>f8d1c78</code></a> Uninstall oclint to ensure gcc can be brew upgraded (<a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1464">#1464</a>)</li>
<li><a href="https://github.com/urllib3/urllib3/commit/cd7cfa613b2678e700597d098ce9bbdc934863e6"><code>cd7cfa6</code></a> Resolve pytest pluggy version conflict (<a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1457">#1457</a>)</li>
<li><a href="https://github.com/urllib3/urllib3/commit/b548abc9812a628d3290d0cab83e44f3c31ac6fe"><code>b548abc</code></a> Update changelog for 1.24 release</li>
<li><a href="https://github.com/urllib3/urllib3/commit/ef0c74542abe69421a86c4d3c6a86fe43cb809a4"><code>ef0c745</code></a> Merging new release version: 1.24</li>
<li><a href="https://github.com/urllib3/urllib3/commit/a0964d9947c07d2b8495726ac23ad251b5d236af"><code>a0964d9</code></a> Add missing key_server_hostname variable (<a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1449">#1449</a>)</li>
<li><a href="https://github.com/urllib3/urllib3/commit/34d8298ecf93c84f9916457d89701d6d5c807780"><code>34d8298</code></a> Test against Python 3.7 on AppVeyor (<a href="https://github-redirect.dependabot.com/urllib3/urllib3/issues/1453">#1453</a>)</li>
<li>Additional commits viewable in <a href="https://github.com/urllib3/urllib3/compare/1.23...1.24.2">compare view</a>
</details>
<br /></li>
</ul>
<p><a href="https://help.github.com/articles/configuring-automated-security-fixes"><img alt="Dependabot compatibility score" src="https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=urllib3&ampSEMICOLONpackage-manager=pip&ampSEMICOLONprevious-version=1.23&ampSEMICOLONnew-version=1.24.2" /></a></p>
<p>Dependabot will resolve any conflicts with this PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting <code>@dependabot rebase</code>.</p>
<hr />
<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot ignore this [patch|minor|major] version` will close this PR and stop Dependabot creating any more for this minor/major version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)
- `@dependabot use these labels` will set the current labels as the default for future PRs for this repo and language
- `@dependabot use these reviewers` will set the current reviewers as the default for future PRs for this repo and language
- `@dependabot use these assignees` will set the current assignees as the default for future PRs for this repo and language
- `@dependabot use this milestone` will set the current milestone as the default for future PRs for this repo and language

You can disable automated security fix PRs for this repo from the [Security Alerts page](https://github.com/atulbipin/spotify-slackbot/network/alerts).

</details>;1571676788000;1571676789000
464977589;464977589;zahin-mohammad;;1562522669000;1562522669000
464977554;464977554;zahin-mohammad;;1562522649000;1562522649000
596177318;596177318;zahin-mohammad;;1586296819000;1586296819000
596168866;596168866;zahin-mohammad;;1586295661000;1586295661000
573408419;573408419;zahin-mohammad;;1583009138000;1583009138000
549249843;549249843;zahin-mohammad;<p>If the script is run within a folder that is not at <code>~/</code> then the wrong path will be assigned from the following lines 
```
    echo "\e[45m Installing zsh-autosuggestions \e[0m"
    # Install zsh-autosuggestions (not via oh-my-zsh)
    git clone https://github.com/zsh-users/zsh-autosuggestions ~/.zsh/zsh-autosuggestions
    echo "source ${(q-)PWD}/.zsh/zsh-autosuggestions/zsh-autosuggestions.zsh" &gtSEMICOLON&gtSEMICOLON ${ZDOTDIR:-$HOME}/.zshrc</p>
<pre><code>echo "\e[45m Installing zsh-syntax-highlighting \e[0m"
# Install zsh-syntax-highlighting (not via oh-my-zsh)
git clone https://github.com/zsh-users/zsh-syntax-highlighting.git ~/.zsh/zsh-syntax-highlighting
echo "source ${(q-)PWD}/.zsh/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh" &gtSEMICOLON&gtSEMICOLON ${ZDOTDIR:-$HOME}/.zshrc
</code></pre>
<p>```</p>;1578959548000;1578959574000
545331007;545331007;zahin-mohammad;;1578170967000;1578170967000
11843860;11843860;AttwellBrian;<p>Clean this up some time. And follow this: Read http://opencv.org/platforms/android/android-best-practices.html before making modifications.</p>;1362861825000;1362861827000
581340665;581340665;dependabot[bot];<p>Bumps <a href="https://github.com/acornjs/acorn">acorn</a> from 5.7.3 to 5.7.4.
<details>
<summary>Commits</summary>
<ul>
<li><a href="https://github.com/acornjs/acorn/commit/6370e90067552022710190319cbbbd8c43001957"><code>6370e90</code></a> Mark version 5.7.4</li>
<li><a href="https://github.com/acornjs/acorn/commit/fbc15b1344f6dfb992f67b4bbf1357436247c8a0"><code>fbc15b1</code></a> More rigorously check surrogate pairs in regexp validator</li>
<li>See full diff in <a href="https://github.com/acornjs/acorn/compare/5.7.3...5.7.4">compare view</a></li>
</ul>
</details>
<br /></p>
<p><a href="https://help.github.com/articles/configuring-automated-security-fixes"><img alt="Dependabot compatibility score" src="https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=acorn&ampSEMICOLONpackage-manager=npm_and_yarn&ampSEMICOLONprevious-version=5.7.3&ampSEMICOLONnew-version=5.7.4" /></a></p>
<p>Dependabot will resolve any conflicts with this PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting <code>@dependabot rebase</code>.</p>
<hr />
<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating it. You can achieve the same result by closing it manually
- `@dependabot ignore this major version` will close this PR and stop Dependabot creating any more for this major version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop Dependabot creating any more for this minor version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)
- `@dependabot use these labels` will set the current labels as the default for future PRs for this repo and language
- `@dependabot use these reviewers` will set the current reviewers as the default for future PRs for this repo and language
- `@dependabot use these assignees` will set the current assignees as the default for future PRs for this repo and language
- `@dependabot use this milestone` will set the current milestone as the default for future PRs for this repo and language

You can disable automated security fix PRs for this repo from the [Security Alerts page](https://github.com/AttwellBrian/nico/network/alerts).

</details>;1584214355000;1584214356000
383727930;383727930;li-boxuan;<p>Hi, this dataset seems interesting, but the latest update is 3 years ago.</p>
<p>Just wondering if there is any plan on updates.</p>;1542959386000;1542959386000
517043197;517043197;anil-goudar;<p>How to specify the vertical limit so that the column contents can be modified accordingly.</p>;1572860880000;1572860880000
481923312;481923312;tanyachawla412;;1566073628000;1566073628000
478103608;478103608;margarineHound;<p>I want the bullet points for each position to line up with the position title and employer. I have modified the \tightitemize command as follows: </p>
<p><code>\newenvironment{tightitemize} % Defines the tightitemize environment which modifies the itemize environment to be more compact
{\vspace{-\topsep}\begin{itemize} \leftmargin-25pt \itemsep0pt \parskip0pt \parsep0pt \itemindent-20pt \labelsep1pt \parindent0pt
}
{\end{itemize}\vspace{-\topsep}}</code></p>
<p>However, it seems that this is causing a hanging indent: ie. I want the second line of each bullet point to line up vertically with the first word in the first line, if not the bullet. </p>
<p><img alt="Rishabh CV_v3 - Online LaTeX Editor Overleaf - Mozilla Firefox 8_7_2019 01_02_38_LI_Moment" src="https://user-images.githubusercontent.com/7854019/62652271-00385400-b929-11e9-961a-42df41e9a32a.jpg" /></p>
<p>I have tried playing around with the values, and parameters, to no avail. I would appreciate any and all help. 
Thank you in advance. </p>;1565206562000;1565206562000
448359411;448359411;hengdashi;;1558733291000;1564500122000
516467450;448359411;AngelOnFira;<p>Thanks for this :100:</p>;1564500122000;1564500122000
412694458;412694458;voidstarstar;<p>Fixes the overlap between the section name and body.
Fixes #42</p>;1550709335000;1550709335000
409262011;409262011;utkuturk;<p>Hello, thank you for your work. It is awesome! Is there a way to write č this character? I could not did it. When I try writing <code>\v{c}</code> it gives a question mark in the place of the character.</p>
<p>Thank you for your work.</p>;1549972576000;1555174481000
482838797;409262011;spilehchiha;<p>For an initial ad-hoc solution, you can try including this command to your preamble:
<code>\usepackage[T1]{fontenc}</code>
Unfortunately, it changes the default font, but it gets the job done and recognizes the upper <code>v</code> above the <code>c</code> character.</p>;1555174481000;1555174481000
350898657;350898657;yadavabhinav16;<p>I tried using this template in both overleaf and shareLaTeX. 
\textbf{ sample text } doesn't seem to work. Anything that I might be missing on?</p>;1534354650000;1540838043000
434028919;350898657;peey;<p>See https://tex.stackexchange.com/a/274901/144456</p>
<p>The solution is to use <code>\custombold{ ... }</code> instead.</p>;1540838043000;1540838043000
318642291;318642291;prateekiiest;<p>Is there any way of modification of colors ?
Since in some areas it kind of gets faded out</p>;1524926755000;1527195519000
391859522;318642291;tonystonee;<p>I agree with this question, I try to change the color of the \section and get error:
<code>Package xcolor Error: Undefined color</code>RED'.`</p>
<p>I can't seem to find a work around, anyone know of one?</p>;1527195519000;1527195519000
307608049;307608049;DeeprajPandey;<p>Added a height argument to \begin{minipage}. When the contents are more than the vertical limit, the extra contents will run into the page instead of overflowing to the next page.
This can be a solution till we figure out a way to smoothly transition to the next page.</p>
<p>Also added a % after the first \end{minipage} which removes the little bit of horizontal space that is automatically inserted by the <code>minipage</code> environment. <a href="https://tex.stackexchange.com/questions/157222/putting-figures-side-by-side-using-minipage">Reference</a></p>;1521718822000;1521718989000
298012184;298012184;srdg;<p>I used this template and while making my CV, I felt it would be a cleaner design if we could separate the two columns with a single <code>vline</code>, with width, say, <code>0.4 pts</code>. Can that be included anywhere?</p>;1518881029000;1518881029000
291973577;291973577;niranjantdesai;<p>I'm using the OpenFonts template on Windows with TeXworks for compilation. There's a problem with the "Publications" section --- the body and the section name are overlapping.</p>
<p><img alt="capture" src="https://user-images.githubusercontent.com/15803433/35452041-45161af8-02eb-11e8-92bb-a6b6f105370d.PNG" /></p>
<p>A better fix seems to be using this:
<code>\renewcommand\refname{PUBLICATIONS}</code>
instead of the current code, which is
<code>\section{Publications}</code>
<code>\renewcommand\refname{\vskip -1.5cm} % Couldn't get this working from the .cls file</code></p>
<p><img alt="capture1" src="https://user-images.githubusercontent.com/15803433/35452188-bcd015c6-02eb-11e8-8d83-c1517ba7d5fd.PNG" /></p>;1516987547000;1550709802000
465815005;291973577;voidstarstar;<p>I didn't see any pull request for this issue, so I created #50.  Thanks for providing the solution.</p>;1550709802000;1550709802000
277975364;277975364;anthonyattard;<p>If you believe this belongs in a separate repo let me know.</p>;1512008341000;1512008341000
273544317;273544317;naveenT1010;<p>Hi,
How can I enable the format to extend into two pages, without first one being blank?
Thanks</p>;1510600425000;1522068507000
376153699;273544317;rohit-upadhya;<p>You can follow what I have done with my CV. I have taken it from this repo only and edited it. 
https://github.com/nocturnalnocturnal/ROHIT_UPADHYA_CV</p>;1522068507000;1522068507000
265499138;265499138;csaund;<p>Right now without any special formatting, any overflow onto a second page is poorly formatted. A version of this which allows experience to take up the full column on the first page and awards, leadership, etc to fully occupy the second would keep the clean style and continue to emphasize work experience. </p>
<p>Have you thought about adapting this to look good on 2 pages? </p>;1507994894000;1545691926000
339554756;265499138;juliocesar-io;<p>Hi @csaund, you can create any pages you want like this (just keep the same <code>textwidth</code>  for the correspondent column)
```
\begin{document}</p>
<p>% PAGE ONE</p>
<p>\begin{minipage}[t]{0.38\textwidth}
% COLUMN ONE SECTIONS
\end{minipage}</p>
<p>\begin{minipage}[t]{0.61\textwidth}
% COLUMN TWO SECTIONS
\end{minipage}</p>
<p>\newpage</p>
<p>% PAGE TWO</p>
<p>\begin{minipage}[t]{0.38\textwidth}
% COLUMN ONE SECTIONS
\end{minipage}</p>
<p>\begin{minipage}[t]{0.61\textwidth}
% COLUMN TWO SECTIONS
\end{minipage}</p>
<p>% PAGE N ...</p>
<p>\end{document}
```</p>;1508995406000;1509310942000
404833506;265499138;arp1561;<p>The above-mentioned way and adding \hfill between mini pages makes the deedy resume work with 2 pages perfectly</p>;1531488523000;1531488523000
449776182;265499138;trenta3;<p>I would also like to suggest the paracol package with which the splitting is much more easy to do without having to adjust the minipages each time you modify the CV. The document would look something like:
```
\begin{document}</p>
<p>\columnratio{0.44}
\begin{paracol}{2}
% LATERAL CONTENT
\switchcolumn
% CENTRAL CONTENT
\end{paracol}
```</p>;1545691926000;1545691926000
261485044;261485044;anthonyattard;<p>Does anyone know of any templates that are stylistically similar to Deedy? </p>;1506639220000;1518661793000
343696950;261485044;wilfredom;<p>Yes-ish: <a href="https://www.sharelatex.com/templates/cover-letters/awesome-cv-cover-letter">AwesomeCV Cover letter</a>.</p>
<p>All instances of red can be changed to black near the beginning of the document: look for the line <code>\colorlet{awesome}{awesome-red}</code> and change <code>red</code> to <code>darknight</code>.</p>
<p>I also tried getting the name displayed in the same font by copying over deedy fonts and changing them in the .cls, but for some reason the last name comes out much bolder. </p>;1510437538000;1510437538000
346216956;261485044;anthonyattard;<p>Thanks for the response.</p>;1511314834000;1511314834000
347975932;261485044;wilfredom;<p>No problem! Another strategy (which I'm testing out right now) is to adapt the <a href="https://www.overleaf.com/latex/templates/single-column-deedy-cv-slash-resume-template/zwyxmkbrfgtz">single-column version</a> of Deedy, by Overleaf.</p>;1511985058000;1511985058000
348061332;261485044;anthonyattard;<p>Checkout my cover letter template https://github.com/anthonyattard/Deedy-Resume</p>;1512008220000;1512008220000
348062553;261485044;wilfredom;<p>Wow, fantastic. Thanks so much, Anthony!</p>;1512008676000;1512008676000
365809119;261485044;anthonyattard;<p>Any chance this can be merged?</p>;1518661793000;1518661793000
254465607;254465607;wrmacrae;<p>See other PR.</p>;1504212998000;1504212998000
254465081;254465081;wrmacrae;<p>I don't think anyone uses "Github://" outside of these resumes. If I'm wrong and this is a real thing, feel free to decline. But it seems weird.</p>;1504212879000;1504212879000
208788231;208788231;joshjacobson;<p>When column 2 and column 1 are the same length (or so they appear, perhaps there is a very small length difference causing this), column 2 causes overflow to the next page, while column 1 does not. </p>
<p>Basically, I am able to have 1 additional line on column 1 than I am on column 2. I would like to adjust it so this last line is allowed in column 2 as well.</p>;1487569123000;1487569123000
190428237;190428237;kfeine;<p>Hi,</p>
<p>It seems there is an issue with small caps in <code>\descript{}</code>: some letters with accents are turned to lowercase instead.</p>
<p><a href="http://hpics.li/5672503">Here is an example</a> with:</p>
<pre><code>\descript{Aeéeèuùaà}
</code></pre>
<p>I don't think it's directly related with Relaway font, since it supports small caps with accents. Maybe it's more an <a href="https://tex.stackexchange.com/questions/286047/tex4ht-getting-accented-letters-in-small-capitals-through-selectfont">issue with selectfont</a>?</p>
<p>Thanks,
Kind regards,</p>;1479505223000;1508523798000
333130640;190428237;reyhanhgh;<p>Hi, 
I have the same problem. Does anybody know how to fix it?</p>
<p>here is the source code: 
https://www.overleaf.com/latex/templates/deedy-resume/sqdbztjjghvz#.Wc4HfBOCzVp</p>
<p>Many thanks!</p>;1506692796000;1506692893000
338285717;190428237;remidebette;<p>Hi everyone,</p>
<p>The problem seems indeed to be due to the Raleway font.
I opened a ticket there a while back detailing the issue: <a href="https://github.com/impallari/Raleway/issues/21">Support for accented small caps</a></p>
<p>Rémi</p>;1508523798000;1508523798000
183199388;183199388;Data-drone;<p>I have tried putting the files in:
<code>~/.fonts/</code>
<code>/usr/local/share/fonts</code>
'/usr/share/fonts/truetype'</p>
<p>I have refreshed the <code>fc-cache</code>.</p>
<p>and <code>fc-list|grep</code> returns the files</p>
<p>but fontspec and kpathea can't find them?</p>;1476523529000;1476523529000
175121549;175121549;jck;;1473105017000;1473143416000
244861096;175121549;nirmitgoyal;<p>@deedy : nice changes. Deedy please merge this pull request</p>;1473143416000;1473143416000
145788671;145788671;alanyee;<p>Under "Availability" in the README.md, the Overleaf link links to a source template that is Version 1.1 (30/4/2014). I think this link should be updated to Version 1.2.</p>;1459800099000;1459800099000
142323881;142323881;kirkang;<p>1)Add xeCJK, it is supported to type Chinese character. Korean and Japanese are also supported theoretical, but it has not been tested.
2)Add new command "the position applied"</p>;1458560666000;1458560666000
127175342;127175342;isometric;<p>I did a spin on your design to simplify the topography and alignment. </p>
<p>See if you like it :)</p>
<p>https://drive.google.com/file/d/0B1mMiFxu14yFdzJIaGtUYlJhUm8/view?usp=docslist_api</p>
<p>Sadly I'm not familiar enough with LaTeX to so it was made it in InDesign but you can find the project file here: </p>
<p>https://drive.google.com/file/d/0B1mMiFxu14yFV0paNVdIaHliaVU/view?usp=docslist_api</p>;1453105203000;1479081239000
172885081;127175342;deedy;<p>I quite like it. I'm no designer, but I think the lowercase headings for each section look nice, although the bold date ranges and location lines seem a little overbearing. The slightly bigger name section looks a bit cleaner but the margins seem a little large around it. I also think the descriptions would look better with bullets so the reader doesnt get lost in a wall of text (even though its paragraphed). Cool stuff though!</p>;1453216743000;1453216743000
260222929;127175342;pratnala;<p>@isometric Your files don't exist anymore?</p>;1479081239000;1479081239000
123607701;123607701;nickyr;<p>This file would go in the OpenFonts directory also I guess.</p>;1450851263000;1450851470000
123607114;123607114;nickyr;<p>Right now, the alignment of employers and positions is a little off. There's an extra space before the "|".</p>
<p>By changing</p>
<p><code>tex
\newcommand{\runsubsection}[1]{\color{subheadings}
\fontspec{Helvetica}\fontsize{12pt}{12pt}\selectfont\bfseries\uppercase {#1} \normalfont}</code></p>
<p>to</p>
<p><code>tex
\newcommand{\runsubsection}[1]{\color{subheadings}
\fontspec{Helvetica}\fontsize{12pt}{12pt}\selectfont\bfseries\uppercase {#1}\normalfont}</code></p>
<p>(removing the space after #1), the extra space goes away.</p>
<p>Before, with underscores for spaces: 
<code>EMPLOYER__|_Position</code>
After: 
<code>EMPLOYER_|_Position</code></p>;1450850711000;1487568984000
280992272;123607114;joshjacobson;<p>I think I actually prefer the two spaces. Perhaps this was an intentional design decision?</p>;1487568984000;1487568984000
92031907;92031907;SquaredPi;<p>I get the following errors:
- Undefined control sequence. [\sectionspace]
- LaTeX Error: Environment tightitemize undefined. [\begin{tightitemize}]</p>
<p>Also, when I try to redefine the tightitemize environment into the document, I get an "already defined environment".</p>
<p>Please advice.
Thank you</p>;1435655287000;1495940774000
131224023;92031907;rhuddles;<p>Having the same issue</p>;1439582798000;1439582798000
143951244;92031907;CorbanR;<p>same issue</p>;1443505429000;1443505429000
276460102;92031907;hawkins;<p>Have any of you tried <code>\sectionsep</code>?</p>;1485889744000;1485889744000
304489671;92031907;MarcoGiancarli;<p>Replace all instances of <code>\sectionspace</code> with <code>\sectionsep</code> and all instances of <code>\tightitemize</code> with <code>\tightemize</code></p>;1495940774000;1495940774000
62188066;62188066;rrshaban;<p>\emph and \textit do nothing within any part of the template, and I haven't been able to figure out why. </p>;1426534802000;1447765627000
157365714;62188066;JanetMatsen;<p>Adding \setmainfont{Lato Light} to the preamble made \textit{} work in the Overleaf Version.  https://www.overleaf.com/latex/templates/deedy-resume/sqdbztjjghvz#.VksmZmQrJcw </p>;1447765616000;1447765616000
57681975;57681975;Void2258;<p>I did my undergrad at the Rochester Institute of Technology. The template puts the name on 2 lines, which is fine, but does so with atrocious spacing, as it spaces out the first three words to occupy the same space as University of Pittsburgh (my grad school). The end result looks horrible, no matter what I do.</p>;1423891489000;1426913614000
84255137;57681975;Travis-S;<p>I had the same problem with my institution, and ended up simply breaking the name apart by hand. Admittedly not an elegant solution, but the alternative is probably a little complicated to code...</p>
<p>It would appear a relevant <code>LaTeX</code> command is the <code>\titlespacing{\subsection}</code> command in the <code>.cls</code> files.</p>;1426913614000;1426913614000
57410171;57410171;EmmanueleSalvati;<p>Update from original</p>;1423709272000;1423709272000
355811190;355811190;brian-norman;<p>Glen requests this visual cue that the level has indeed changes, a good idea!</p>;1535680080000;1535994770000
585591514;585591514;brian-norman;;1584827209000;1584828545000
602111660;585591514;brian-norman;<p>lol</p>;1584828544000;1584828544000
585583679;585583679;zahin-mohammad;;1584824397000;1585023151000
603003326;585583679;zahin-mohammad;<p>Test</p>;1585023084000;1585023084000
603003391;585583679;zahin-mohammad;<blockquote>
<p>Test</p>
</blockquote>
<p>Yo this is a reply</p>;1585023100000;1585023100000
603003447;585583679;zahin-mohammad;<p>Comment 1</p>;1585023113000;1585023113000
603003468;585583679;zahin-mohammad;<p>Comment 2</p>;1585023118000;1585023118000
603003551;585583679;zahin-mohammad;<blockquote>
<p>Comment 1</p>
</blockquote>
<p>Reply to 1</p>
<blockquote>
<p>Comment 2</p>
</blockquote>
<p>Reply to comment 2</p>;1585023137000;1585023137000
603003605;585583679;zahin-mohammad;<p>Multiline
Comment</p>;1585023151000;1585023151000
357461207;357461207;brian-norman;<p>Brian this would be significantly cooler if it auto-ran on a fixed time-period as you originally imagined... Learn how to do that! Heroku maybe?</p>;1536197030000;1536197030000
451217561;451217561;brian-norman;;1559498415000;1559498415000
356879856;356879856;brian-norman;<p>Or any user's commits that week.</p>;1536077319000;1554088072000
355453005;355453005;brian-norman;<p>!google “anything”</p>;1535613056000;1535613570000
355452779;355452779;brian-norman;<p>!git issues user repo add "Issue title" "Issue description"</p>;1535613006000;1535613570000
383953069;383953069;keenangaudio;;1543027545000;1543027545000
383948347;383948347;keenangaudio;;1543022138000;1543022138000
383948004;383948004;louisj381;<p>fudge</p>;1543021796000;1543021796000
536209332;536209332;SunnyNagam;;1576052051000;1576052051000
598240001;598240001;nakulkhambhati;<p>I tried to open the example scene WriteStuff, following the youtube tutorial (https://www.youtube.com/watch?time_continue=492&ampSEMICOLONv=ZltiKHFWmv8&ampSEMICOLONfeature=emb_logo)
it worked for SquareToCircle but not for WriteStuff which tells me that latex is not working. I think it may have something to do with my path variables but im not too sure how they work. I have installed miktex only for my user, so i only changed user variables and not system variables. This is my path variable: C:\Users\Nakul Khambhati\AppData\Local\Programs\Python\Python37\Scripts\SEMICOLONC:\Users\Nakul Khambhati\AppData\Local\Programs\Python\Python37\SEMICOLONC:\MiKTeX 2.9\miktex\bin\x64\SEMICOLONC:\FFmpeg\binSEMICOLONC:\sox-14-4-2SEMICOLON</p>
<p>I tried reinstalling Miktex too but nothing seems to work. Can someone please help? I am interested in coding math videos. </p>;1586601615000;1586601955000
612390966;598240001;Elteoremadebeethoven;<p>There are a lot of problems like yours, both here on <a href="https://github.com/3b1b/manim/issues?q=latex+error">Github</a> and <a href="https://stackoverflow.com/search?q=%5Bmanim%5D+latex">Stackoverflow</a>, check them out.</p>;1586601955000;1586601955000
597818092;597818092;Mathoholic;<p>`Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.</p>
<p>c:\manim&gtSEMICOLONpython -m pip install -r requirements.txt
Ignoring pycairo: markers 'sys_platform == "linux"' don't match your environment</p>
<p>Requirement already satisfied: argparse==1.4.0 in c:\program files\python37-32\l
ib\site-packages (from -r requirements.txt (line 1)) (1.4.0)
Requirement already satisfied: colour==0.1.5 in c:\program files\python37-32\lib
\site-packages (from -r requirements.txt (line 2)) (0.1.5)
Requirement already satisfied: numpy==1.16.4 in c:\program files\python37-32\lib
\site-packages (from -r requirements.txt (line 3)) (1.16.4)
Requirement already satisfied: Pillow==5.2.0 in c:\program files\python37-32\lib
\site-packages (from -r requirements.txt (line 4)) (5.2.0)
Requirement already satisfied: progressbar==2.5 in c:\program files\python37-32\
lib\site-packages (from -r requirements.txt (line 5)) (2.5)
Requirement already satisfied: scipy==1.3.0 in c:\program files\python37-32\lib\
site-packages (from -r requirements.txt (line 6)) (1.3.0)
Requirement already satisfied: tqdm==4.24.0 in c:\program files\python37-32\lib\
site-packages (from -r requirements.txt (line 7)) (4.24.0)
Requirement already satisfied: opencv-python==3.4.2.17 in c:\program files\pytho
n37-32\lib\site-packages (from -r requirements.txt (line 8)) (3.4.2.17)
Collecting pycairo&gtSEMICOLON=1.18.1
  Using cached pycairo-1.19.1.tar.gz (205 kB)
Collecting pydub==0.23.0
  Using cached pydub-0.23.0-py2.py3-none-any.whl (28 kB)
Processing c:\users\wayne tower\appdata\local\pip\cache\wheels\00\6e\d4\7c4b7bc2
2c090baf4f470e7f35c4f22206277229bdb6607df6\pyreadline-2.1-py3-none-any.whl
Building wheels for collected packages: pycairo
  Building wheel for pycairo (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Program Files\Python37-32\python.exe' -u -c 'import sys, setupto
ols, tokenizeSEMICOLON sys.argv[0] = '"'"'C:\Users\WAYNET~1\AppData\Local\Temp\pip
-install-172ou340\pycairo\setup.py'"'"'SEMICOLON <strong>file</strong>='"'"'C:\Users\WAYNET~1\Ap
pData\Local\Temp\pip-install-172ou340\pycairo\setup.py'"'"'SEMICOLONf=getattr(token
ize, '"'"'open'"'"', open)(<strong>file</strong>)SEMICOLONcode=f.read().replace('"'"'\r\n'"'"', '"'"'
\n'"'"')SEMICOLONf.close()SEMICOLONexec(compile(code, <strong>file</strong>, '"'"'exec'"'"'))' bdist_wheel -d
 'C:\Users\WAYNET~1\AppData\Local\Temp\pip-wheel-0yqjrek4'
       cwd: C:\Users\WAYNET~1\AppData\Local\Temp\pip-install-172ou340\pycairo\
  Complete output (18 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win32-3.7
  creating build\lib.win32-3.7\cairo
  copying cairo__init__.py -&gtSEMICOLON build\lib.win32-3.7\cairo
  copying cairo__init__.pyi -&gtSEMICOLON build\lib.win32-3.7\cairo
  copying cairo\py.typed -&gtSEMICOLON build\lib.win32-3.7\cairo
  running build_ext
  building 'cairo._cairo' extension
  creating build\temp.win32-3.7
  creating build\temp.win32-3.7\Release
  creating build\temp.win32-3.7\Release\cairo
  C:\Program Files\Microsoft Visual Studio 14.0\VC\BIN\cl.exe /c /nologo /Ox /W3
 /GL /DNDEBUG /MD -DPYCAIRO_VERSION_MAJOR=1 -DPYCAIRO_VERSION_MINOR=19 -DPYCAIRO
_VERSION_MICRO=1 "-IC:\Program Files\Python37-32\include" "-IC:\Program Files\Py
thon37-32\include" "-IC:\Program Files\Microsoft Visual Studio 14.0\VC\INCLUDE"
"-IC:\Program Files\Windows Kits\10\include\10.0.10240.0\ucrt" "-IC:\Program Fil
es\Windows Kits\8.1\include\shared" "-IC:\Program Files\Windows Kits\8.1\include
\um" "-IC:\Program Files\Windows Kits\8.1\include\winrt" /Tccairo/device.c /Fobu
ild\temp.win32-3.7\Release\cairo/device.obj
  device.c
  c:\users\wayne tower\appdata\local\temp\pip-install-172ou340\pycairo\cairo\pyc
airo.h(37): fatal error C1083: Cannot open include file: 'cairo.h': No such file
 or directory
  error: command 'C:\Program Files\Microsoft Visual Studio 14.0\VC\BIN\cl.e
xe' failed with exit status 2</p>
<hr />
<p>ERROR: Failed building wheel for pycairo
  Running setup.py clean for pycairo
Failed to build pycairo
Installing collected packages: pycairo, pydub, pyreadline
    Running setup.py install for pycairo ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Program Files\Python37-32\python.exe' -u -c 'import sys, setup
tools, tokenizeSEMICOLON sys.argv[0] = '"'"'C:\Users\WAYNET~1\AppData\Local\Temp\p
ip-install-172ou340\pycairo\setup.py'"'"'SEMICOLON <strong>file</strong>='"'"'C:\Users\WAYNET~1\
AppData\Local\Temp\pip-install-172ou340\pycairo\setup.py'"'"'SEMICOLONf=getattr(tok
enize, '"'"'open'"'"', open)(<strong>file</strong>)SEMICOLONcode=f.read().replace('"'"'\r\n'"'"', '"'
"'\n'"'"')SEMICOLONf.close()SEMICOLONexec(compile(code, <strong>file</strong>, '"'"'exec'"'"'))' install --re
cord 'C:\Users\WAYNET~1\AppData\Local\Temp\pip-record-wa2qfl0f\install-record.tx
t' --single-version-externally-managed --compile --install-headers 'C:\Program F
iles\Python37-32\Include\pycairo'
         cwd: C:\Users\WAYNET~1\AppData\Local\Temp\pip-install-172ou340\pycairo\</p>
<pre><code>Complete output (18 lines):
running install
running build
running build_py
creating build
creating build\lib.win32-3.7
creating build\lib.win32-3.7\cairo
copying cairo\__init__.py -&gtSEMICOLON build\lib.win32-3.7\cairo
copying cairo\__init__.pyi -&gtSEMICOLON build\lib.win32-3.7\cairo
copying cairo\py.typed -&gtSEMICOLON build\lib.win32-3.7\cairo
running build_ext
building 'cairo._cairo' extension
creating build\temp.win32-3.7
creating build\temp.win32-3.7\Release
creating build\temp.win32-3.7\Release\cairo
C:\Program Files\Microsoft Visual Studio 14.0\VC\BIN\cl.exe /c /nologo /Ox /
</code></pre>
<p>W3 /GL /DNDEBUG /MD -DPYCAIRO_VERSION_MAJOR=1 -DPYCAIRO_VERSION_MINOR=19 -DPYCAI
RO_VERSION_MICRO=1 "-IC:\Program Files\Python37-32\include" "-IC:\Program Files\
Python37-32\include" "-IC:\Program Files\Microsoft Visual Studio 14.0\VC\INCLUDE
" "-IC:\Program Files\Windows Kits\10\include\10.0.10240.0\ucrt" "-IC:\Program F
iles\Windows Kits\8.1\include\shared" "-IC:\Program Files\Windows Kits\8.1\inclu
de\um" "-IC:\Program Files\Windows Kits\8.1\include\winrt" /Tccairo/device.c /Fo
build\temp.win32-3.7\Release\cairo/device.obj
    device.c
    c:\users\wayne tower\appdata\local\temp\pip-install-172ou340\pycairo\cairo\p
ycairo.h(37): fatal error C1083: Cannot open include file: 'cairo.h': No such fi
le or directory
    error: command 'C:\Program Files\Microsoft Visual Studio 14.0\VC\BIN\cl
.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Program Files\Python37-32\pyt
hon.exe' -u -c 'import sys, setuptools, tokenizeSEMICOLON sys.argv[0] = '"'"'C:\Users\
WAYNET~1\AppData\Local\Temp\pip-install-172ou340\pycairo\setup.py'"'"'SEMICOLON <strong>
file</strong>='"'"'C:\Users\WAYNET~1\AppData\Local\Temp\pip-install-172ou340\pyc
airo\setup.py'"'"'SEMICOLONf=getattr(tokenize, '"'"'open'"'"', open)(<strong>file</strong>)SEMICOLONcode=f.r
ead().replace('"'"'\r\n'"'"', '"'"'\n'"'"')SEMICOLONf.close()SEMICOLONexec(compile(code, __file_
_, '"'"'exec'"'"'))' install --record 'C:\Users\WAYNET~1\AppData\Local\Temp\pip-
record-wa2qfl0f\install-record.txt' --single-version-externally-managed --compil
e --install-headers 'C:\Program Files\Python37-32\Include\pycairo' Check the log
s for full command output.</p>
<p>c:\manim&gtSEMICOLON`</p>;1586514250000;1586514250000
596764319;596764319;xvetisky;<p>Hi. When I installed Manim, I first tried to render an animation to see if it works. I first issued the command below to render the <em>Square to Circle</em> example, and it did well.
<code>python3 ./manim.py example_scenes.py SquareToCircle -pl</code>
But, whenever I try to render the other examples provided, it always makes the same error.
<code>Exception: Latex error converting to dvi. See log output above or the log file: media/Tex/664e80e8d8d98482.log</code>
Command issued : <code>sudo -H python3 -m manim example_scenes.py OpeningManimExample -pl</code>
This is the complete console log : 
```
Media will be written to ./media/. You can change this behavior with the --media_dir flag.
latex: warning: running with administrator privileges</p>
<p>Sorry, but latex did not succeed.</p>
<p>The log file hopefully contains the information to get MiKTeX going again:</p>
<p>/var/root/Library/Application Support/MiKTeX/texmfs/data/miktex/log/latex.log</p>
<p>Traceback (most recent call last):
  File "/Users/mac/Downloads/manim-master/manimlib/extract_scene.py", line 155, in main
    scene = SceneClass(**scene_kwargs)
  File "/Users/mac/Downloads/manim-master/manimlib/scene/scene.py", line 53, in <strong>init</strong>
    self.construct()
  File "example_scenes.py", line 20, in construct
    title = TextMobject("This is some \LaTeX")
  File "/Users/mac/Downloads/manim-master/manimlib/mobject/svg/tex_mobject.py", line 147, in <strong>init</strong>
    SingleStringTexMobject.<strong>init</strong>(
  File "/Users/mac/Downloads/manim-master/manimlib/mobject/svg/tex_mobject.py", line 42, in <strong>init</strong>
    file_name = tex_to_svg_file(
  File "/Users/mac/Downloads/manim-master/manimlib/utils/tex_file_writing.py", line 20, in tex_to_svg_file
    dvi_file = tex_to_dvi(tex_file)
  File "/Users/mac/Downloads/manim-master/manimlib/utils/tex_file_writing.py", line 70, in tex_to_dvi
    raise Exception(
Exception: Latex error converting to dvi. See log output above or the log file: media/Tex/664e80e8d8d98482.log
```
I tried to search on google for any help, but all the that I read was about the same issue but on Windows, and I'm on mac. However, I continued searching and found this on StackOverflow : </p>
<blockquote>
<p>That answer is no longer useful now because Manim has already solved that problem. So this may be due to two things: To give us a hint of what is wrong, go to the manimlib folder with the terminal and run the following:</p>
<p><code>latex tex_template.tex</code>
We distinguish the two possible errors:</p>
<p>Recognized the latex command but the compilation could not be completed because you are missing libraries, to solve this you can install the missing packages indicated by the terminal.</p>
<p>That LaTeX is not recognized as a command, in this case surely the PATH variable is not set correctly, here is the solution.</p>
</blockquote>
<p>So I went to the manimlib folder and issued the following : <code>latex tex_template.tex</code>. And judging by the console log, it doesn't seem like its the second option because the latex command was recognized, but there is no further information about what's missing. This is the complete console log : 
```
MacBook-Pro-de-MAC:manimlib mac$ latex tex_template.tex
log4cxx: setFile(/Users/mac/Library/Application Support/MiKTeX/texmfs/data/miktex/log/latex.log,true) call failed.
log4cxx: IO Exception : status code = 13
log4cxx: No output stream or file set for the appender named [RollingLogFile].
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (MiKTeX 2.9.7250)</p>
<p>Sorry, but latex did not succeed.</p>
<p>The log file hopefully contains the information to get MiKTeX going again:</p>
<p>/Users/mac/Library/Application Support/MiKTeX/texmfs/data/miktex/log/latex.log
MacBook-Pro-de-MAC:manimlib mac$ latex tex_template.tex
log4cxx: setFile(/Users/mac/Library/Application Support/MiKTeX/texmfs/data/miktex/log/latex.log,true) call failed.
log4cxx: IO Exception : status code = 13
log4cxx: No output stream or file set for the appender named [RollingLogFile].
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (MiKTeX 2.9.7250)</p>
<p>Sorry, but latex did not succeed.</p>
<p>The log file hopefully contains the information to get MiKTeX going again:</p>
<p>/Users/mac/Library/Application Support/MiKTeX/texmfs/data/miktex/log/latex.log
MacBook-Pro-de-MAC:manimlib mac$ latex tex_template.tex
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (MiKTeX 2.9.7250)</p>
<p>Sorry, but latex did not succeed.</p>
<p>The log file hopefully contains the information to get MiKTeX going again:</p>
<p>/Users/mac/Library/Application Support/MiKTeX/texmfs/data/miktex/log/latex.log
```</p>
<p>So I went to see what's on the <code>latex.log</code> file, and found this : </p>
<p>```
2020-01-28 21:28:52,077+0100 INFO  texworks - starting: MiKTeX TeXworks 0.6.3 (MiKTeX 2.9.7250)
2020-04-08 17:31:28,130+0100 INFO  latex - starting with command line: latex tex_template.tex
2020-04-08 17:31:28,135+0100 INFO  latex - allowing known shell commands
2020-04-08 17:31:28,135+0100 INFO  latex - enabling input (output) from (to) processes
2020-04-08 17:31:28,151+0100 INFO  latex - going to create file: latex.fmt
2020-04-08 17:31:29,772+0100 ERROR latex - /usr/local/bin/initexmf did not succeedSEMICOLON exitCode: 1
2020-04-08 17:31:29,772+0100 ERROR latex - output:
2020-04-08 17:31:29,772+0100 ERROR latex - log4cxx: setFile(/Users/mac/Library/Application Support/MiKTeX/texmfs/data/miktex/log/initexmf.log,true) call failed.
log4cxx: IO Exception : status code = 13
log4cxx: No output stream or file set for the appender named [RollingLogFile].</p>
<p>Sorry, but "MiKTeX Configuration Utility" did not succeed.</p>
<p>The log file hopefully contains the information to get MiKTeX going again:</p>
<p>/Users/mac/Library/Application Support/MiKTeX/texmfs/data/miktex/log/initexmf.log</p>
<p>2020-04-08 17:31:29,774+0100 FATAL latex - The memory dump file could not be found.
2020-04-08 17:31:29,774+0100 INFO  latex - finishing with exit code 1
```</p>
<p>But I have no clue what to do. So please Help me. And also, I'm sorry for my probably bad english, I'm french so, be kind ^^</p>;1586369563000;1586391992000
611119595;596764319;xvetisky;<p>Got some other information on the <code>initexmf.log</code>file :
```
020-04-08 19:28:22,695+0100 INFO  initexmf - starting: MiKTeX Configuration Utility 2.9.7259 (MiKTeX 2.9.7250)
2020-04-08 19:28:22,697+0100 INFO  initexmf - Operating on the private (per-user) MiKTeX setup
2020-04-08 19:28:22,697+0100 INFO  initexmf - starting with command line: initexmf --dump-by-name=latex --engine=pdftex
2020-04-08 19:28:22,699+0100 INFO  initexmf - running: miktex-makefmt --engine=pdftex --dest-name=latex --no-dump latex.ini --engine-option=-tcx=cp227.tcx --miktex-disable-maintenance
2020-04-08 19:28:23,144+0100 FATAL initexmf - Permission denied: path="/Users/mac/Library/Application Support/MiKTeX/texmfs/data/miktex/log/miktex-makefmt_2020-04-08-192823.out"</p>
<p>```</p>;1586370628000;1586370628000
611261350;596764319;liyiliuxingyu;<p>只需要对texlive进行更新即可
tlmgr update --self --all --reinstall-forcibly-removed</p>;1586391992000;1586391992000
596128505;596128505;Yoannson;<p>Hello,
I manage to install almost everything with a manim tutorial by "theorem of Beethoven" on Windows 10
https://www.youtube.com/watch?v=ZltiKHFWmv8
Everything went fine, and at the end, the first test "transform square in circle" works fine, but the LaTeX test went wrong. 
<img alt="Capture1" src="https://user-images.githubusercontent.com/63317875/78715518-ce471980-791d-11ea-8330-e9d7b63da791.PNG" />
I don't know what to do, I don't understand what's going on.
If anyone can help, I would really appreciate.
Thanks for all,
Yoann</p>;1586290894000;1586505836000
610685650;596128505;Elteoremadebeethoven;<p>Please read what your terminal says, it is understood that the <code>dvisvgm</code> command is not recognized as an internal command, that is, it was not included when you installed LaTeX, that could happen for two things, or you only installed the basic version or there was an error in the installation, investigate how to install that dependency in some forum.</p>;1586305753000;1586305753000
611009540;596128505;Yoannson;<p>I use LaTeX on a daily basis, I had probably an old version of MikTex. I unsinstalled and reinstalled it, everything works fine now. Problem solved.
Thanks for the help !</p>;1586357986000;1586358018000
611927523;596128505;Elteoremadebeethoven;<p>Ok, please close the issue</p>;1586505836000;1586505836000
595963657;595963657;emailtowa;<p>Hello,</p>
<p>I was looking for a way to plot the tangent (small sectant) at a point and then move it along the graph.
I could succeed with the first part but could not move it (and I tried to find it in the old projects of eoc).</p>
<p>Maybe there is a better way to plot the tangent - not via sectant as below.</p>
<p>Many thanks,
Thomas</p>
<p>from manimlib.imports import *</p>
<p>NEW_BLUE = "#68a8e1"</p>
<p>class v(GraphScene):
    CONFIG = {
        "y_max": 8,
        "y_axis_height": 5,
    }</p>
<pre><code>def construct(self):
    self.show_function_graph()

def show_function_graph(self):
    self.setup_axes(animate=False)
    def func(x):
        return 0.1 * (x + 3-5) * (x - 3-5) * (x-5) + 5

    def rect(x):
        return 2.775*(x-1.5)+3.862
    recta = self.get_graph(rect,x_min=-1,x_max=5)
    graph = self.get_graph(func,x_min=0.2,x_max=9)
    graph.set_color(NEW_BLUE)
    input_tracker_p1 = ValueTracker(1.5)
    input_tracker_p2 = ValueTracker(2.5) #3.5 was original coordinate

    def get_x_value(input_tracker):
        return input_tracker.get_value()

    def get_y_value(input_tracker):
        return graph.underlying_function(get_x_value(input_tracker))

    def get_x_point(input_tracker):
        return self.coords_to_point(get_x_value(input_tracker), 0)

    def get_y_point(input_tracker):
        return self.coords_to_point(0, get_y_value(input_tracker))

    def get_graph_point(input_tracker):
        return self.coords_to_point(get_x_value(input_tracker), get_y_value(input_tracker))

    def get_v_line(input_tracker):
        return DashedLine(get_x_point(input_tracker), get_graph_point(input_tracker), stroke_width=2)

    def get_h_line(input_tracker):
        return DashedLine(get_graph_point(input_tracker), get_y_point(input_tracker), stroke_width=2)
    # 
    input_triangle_p1 = RegularPolygon(n=3, start_angle=TAU / 4)
    output_triangle_p1 = RegularPolygon(n=3, start_angle=0)
    for triangle in input_triangle_p1, output_triangle_p1:
        triangle.set_fill(WHITE, 1)
        triangle.set_stroke(width=0)
        triangle.scale(0.1)
    # 
    input_triangle_p2 = RegularPolygon(n=3, start_angle=TAU / 4)
    output_triangle_p2 = RegularPolygon(n=3, start_angle=0)
    for triangle in input_triangle_p2, output_triangle_p2:
        triangle.set_fill(WHITE, 1)
        triangle.set_stroke(width=0)
        triangle.scale(0.1)

    # 
    x_label_p1 = TexMobject("a")
    output_label_p1 = TexMobject("f(a)")
    x_label_p2 = TexMobject("b")
    output_label_p2 = TexMobject("f(b)")
    v_line_p1 = get_v_line(input_tracker_p1)
    v_line_p2 = get_v_line(input_tracker_p2)
    h_line_p1 = get_h_line(input_tracker_p1)
    h_line_p2 = get_h_line(input_tracker_p2)
    graph_dot_p1 = Dot(color=WHITE)
    graph_dot_p2 = Dot(color=WHITE)

    # reposition mobjects
    x_label_p1.next_to(v_line_p1, DOWN)
    x_label_p2.next_to(v_line_p2, DOWN)
    output_label_p1.next_to(h_line_p1, LEFT)
    output_label_p2.next_to(h_line_p2, LEFT)
    input_triangle_p1.next_to(v_line_p1, DOWN, buff=0)
    input_triangle_p2.next_to(v_line_p2, DOWN, buff=0)
    output_triangle_p1.next_to(h_line_p1, LEFT, buff=0)
    output_triangle_p2.next_to(h_line_p2, LEFT, buff=0)
    graph_dot_p1.move_to(get_graph_point(input_tracker_p1))
    graph_dot_p2.move_to(get_graph_point(input_tracker_p2))


    #
    self.play(
        ShowCreation(graph),
    )
    # Animacion del punto a
    self.add_foreground_mobject(graph_dot_p1)
    self.add_foreground_mobject(graph_dot_p2)
    self.play(
        DrawBorderThenFill(input_triangle_p1),
        Write(x_label_p1),
        ShowCreation(v_line_p1),
        GrowFromCenter(graph_dot_p1),
        ShowCreation(h_line_p1),
        Write(output_label_p1),
        DrawBorderThenFill(output_triangle_p1),
        DrawBorderThenFill(input_triangle_p2),
        Write(x_label_p2),
        ShowCreation(v_line_p2),
        GrowFromCenter(graph_dot_p2),
        ShowCreation(h_line_p2),
        Write(output_label_p2),
        DrawBorderThenFill(output_triangle_p2),
        run_time=0.5
    )
    self.add(
        input_triangle_p2,
        x_label_p2,
        graph_dot_p2,
        v_line_p2,
        h_line_p2,
        output_triangle_p2,
        output_label_p2,
    )
    ###################
    pendiente_recta = self.get_secant_slope_group(
        1.9, recta, dx = 1.4,
        df_label = None,
        dx_label = None,
        dx_line_color = PURPLE,
        df_line_color= ORANGE,
        )
    grupo_secante = self.get_secant_slope_group(
        1.5, graph, dx = 0.001,
        df_label = None,
        dx_label = None,
        dx_line_color = "#942357",
        df_line_color= "#3f7d5c",
        secant_line_color = RED,
    )


    self.add(
        input_triangle_p2,
        graph_dot_p2,
        v_line_p2,
        h_line_p2,
        output_triangle_p2,
    )
    self.play(FadeIn(grupo_secante))





    grupo_secante2 = self.get_secant_slope_group(
        3.5, graph, dx = 0.001,
        df_label = None,
        dx_label = None,
        dx_line_color = "#942357",
        df_line_color= "#3f7d5c",
        secant_line_color = RED,
    )
    self.add(
        input_triangle_p2,
        graph_dot_p2,
        v_line_p2,
        h_line_p2,
        output_triangle_p2,
    )
    self.play(FadeIn(grupo_secante2))


    kwargs = {
        "x_min" : 4,
        "x_max" : 9,
        "fill_opacity" : 0.75,
        "stroke_width" : 0.25,
    }
    self.graph=graph
    iteraciones=6


    self.rect_list = self.get_riemann_rectangles_list(
        graph, iteraciones,start_color=PURPLE,end_color=ORANGE, **kwargs
    )
    flat_rects = self.get_riemann_rectangles(
        self.get_graph(lambda x : 0), dx = 0.5,start_color=invert_color(PURPLE),end_color=invert_color(ORANGE),**kwargs
    )
    rects = self.rect_list[0]
    self.transform_between_riemann_rects(
        flat_rects, rects, 
        replace_mobject_with_target_in_scene = True,
        run_time=0.9
    )

    # adding manim
    picture = Group(*self.mobjects)
    picture.scale(0.6).to_edge(LEFT, buff=SMALL_BUFF)
    manim = TextMobject("Manim").set_height(1.5) \
                                .next_to(picture, RIGHT) \
                                .shift(DOWN * 0.7)
    self.add(manim)
</code></pre>;1586273973000;1586570728000
610647329;595963657;zavden;<p>That scene is easy to do if you understand the update functions, you have to see <a href="https://www.youtube.com/watch?v=l4B7IH1D1uI&ampSEMICOLONlist=PL2B6OzTsMUrwo4hA3BBfS7ZR34K361Z8F&ampSEMICOLONindex=28">8.1</a> and <a href="https://www.youtube.com/watch?v=DnWMwkcZGkM&ampSEMICOLONlist=PL2B6OzTsMUrwo4hA3BBfS7ZR34K361Z8F&ampSEMICOLONindex=29">8.2</a> until the end, and the <a href="https://www.youtube.com/watch?v=-KYMrwhLiV8&ampSEMICOLONlist=PL2B6OzTsMUrwo4hA3BBfS7ZR34K361Z8F&ampSEMICOLONindex=34">examples</a>.</p>;1586297766000;1586297766000
610955310;595963657;emailtowa;<p>zavden, many thanks, very useful tutorials of you.</p>
<p>After watching some of them I had the following idea, it is a quick win approximation to the moving slope. Actually it looks kind of rolling - compare code below:</p>
<p>from manimlib.imports import *</p>
<p>NEW_BLUE = "#68a8e1"</p>
<p>class Play(GraphScene):
    CONFIG = {
        "y_max": 8,
        "y_axis_height": 5,
    }</p>
<pre><code>def construct(self):
    self.show_function_graph()

def show_function_graph(self):
    self.setup_axes(animate=False)
    def func(x):
        return 0.1 * (x + 3-5) * (x - 3-5) * (x-5) + 5

    def rect(x):
        return 2.775*(x-1.5)+3.862
    recta = self.get_graph(rect,x_min=-1,x_max=5)
    graph = self.get_graph(func,x_min=0.2,x_max=9)
    graph.set_color(NEW_BLUE)
    input_tracker_p1 = ValueTracker(1.5)
    input_tracker_p2 = ValueTracker(2.5) #3.5 was original coordinate

    def get_x_value(input_tracker):
        return input_tracker.get_value()

    def get_y_value(input_tracker):
        return graph.underlying_function(get_x_value(input_tracker))

    def get_x_point(input_tracker):
        return self.coords_to_point(get_x_value(input_tracker), 0)

    def get_y_point(input_tracker):
        return self.coords_to_point(0, get_y_value(input_tracker))

    def get_graph_point(input_tracker):
        return self.coords_to_point(get_x_value(input_tracker), get_y_value(input_tracker))

    def get_v_line(input_tracker):
        return DashedLine(get_x_point(input_tracker), get_graph_point(input_tracker), stroke_width=2)

    def get_h_line(input_tracker):
        return DashedLine(get_graph_point(input_tracker), get_y_point(input_tracker), stroke_width=2)
    # 
    input_triangle_p1 = RegularPolygon(n=3, start_angle=TAU / 4)
    output_triangle_p1 = RegularPolygon(n=3, start_angle=0)
    for triangle in input_triangle_p1, output_triangle_p1:
        triangle.set_fill(WHITE, 1)
        triangle.set_stroke(width=0)
        triangle.scale(0.1)
    # 
    input_triangle_p2 = RegularPolygon(n=3, start_angle=TAU / 4)
    output_triangle_p2 = RegularPolygon(n=3, start_angle=0)
    for triangle in input_triangle_p2, output_triangle_p2:
        triangle.set_fill(WHITE, 1)
        triangle.set_stroke(width=0)
        triangle.scale(0.1)

    # 
    x_label_p1 = TexMobject("a")
    output_label_p1 = TexMobject("f(a)")
    x_label_p2 = TexMobject("b")
    output_label_p2 = TexMobject("f(b)")
    v_line_p1 = get_v_line(input_tracker_p1)
    v_line_p2 = get_v_line(input_tracker_p2)
    h_line_p1 = get_h_line(input_tracker_p1)
    h_line_p2 = get_h_line(input_tracker_p2)
    graph_dot_p1 = Dot(color=WHITE)
    graph_dot_p2 = Dot(color=WHITE)

    # reposition mobjects
    x_label_p1.next_to(v_line_p1, DOWN)
    x_label_p2.next_to(v_line_p2, DOWN)
    output_label_p1.next_to(h_line_p1, LEFT)
    output_label_p2.next_to(h_line_p2, LEFT)
    input_triangle_p1.next_to(v_line_p1, DOWN, buff=0)
    input_triangle_p2.next_to(v_line_p2, DOWN, buff=0)
    output_triangle_p1.next_to(h_line_p1, LEFT, buff=0)
    output_triangle_p2.next_to(h_line_p2, LEFT, buff=0)
    graph_dot_p1.move_to(get_graph_point(input_tracker_p1))
    graph_dot_p2.move_to(get_graph_point(input_tracker_p2))

    self.play(
        ShowCreation(graph),
    )


    n = 25
    list1 = [None] * (n+1)
    for i in range(0,n+1):
        list1[i] = self.get_secant_slope_group(
            3.5-2*i/n, graph, dx = 0.001,
            df_label = None,
            dx_label = None,
            dx_line_color = "#942357",
            df_line_color= "#3f7d5c",
            secant_line_color = RED,
        )

    for j in range(n-1):
        self.remove(list1[j])
        self.add(list1[j+1])
        self.wait(0.1)
</code></pre>;1586352024000;1586352024000
612297570;595963657;zhuzihao-hz;<p>I have similar problems, the update functions help me a lot, Thank you very much. 👍 @zavden </p>;1586570728000;1586570728000
595268372;595268372;ashish453;<p>i did this and get this :</p>
<p>C:\Manim\manim-master&gtSEMICOLONpython -m manim example_scenes.py SquareToCircle -pl
Traceback (most recent call last):
  File "C:\Python38-32\lib\runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python38-32\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Manim\manim-master\manim.py", line 2, in <module>
    import manimlib
  File "C:\Manim\manim-master\manimlib__init__.py", line 4, in <module>
    import manimlib.extract_scene
  File "C:\Manim\manim-master\manimlib\extract_scene.py", line 9, in <module>
    from manimlib.scene.scene import Scene
  File "C:\Manim\manim-master\manimlib\scene\scene.py", line 9, in <module>
    from manimlib.animation.animation import Animation
  File "C:\Manim\manim-master\manimlib\animation\animation.py", line 5, in <module>
    from manimlib.mobject.mobject import Mobject
  File "C:\Manim\manim-master\manimlib\mobject\mobject.py", line 15, in <module>
    from manimlib.utils.color import color_gradient
  File "C:\Manim\manim-master\manimlib\utils\color.py", line 8, in <module>
    from manimlib.utils.bezier import interpolate
  File "C:\Manim\manim-master\manimlib\utils\bezier.py", line 1, in <module>
    from scipy import linalg
  File "C:\Python38-32\lib\site-packages\scipy\linalg__init__.py", line 195, in <module>
    from .misc import *
  File "C:\Python38-32\lib\site-packages\scipy\linalg\misc.py", line 5, in <module>
    from .blas import get_blas_funcs
  File "C:\Python38-32\lib\site-packages\scipy\linalg\blas.py", line 215, in <module>
    from scipy.linalg import _fblas
ImportError: DLL load failed while importing _fblas: The specified module could not be found.</p>;1586191800000;1586191800000
595237885;595237885;yutharsan;<h3>If this is a support request:</h3>
<p><strong>Please attempt to solve the problem on your own before opening an issue.</strong>
Between old issues, StackOverflow, and Google, you should be able to find
solutions to most of the common problems.</p>
<p>Include at least:
1. Steps to reproduce the issue (e.g. the command you ran)
2. The unexpected behavior that occurred (e.g. error messages or screenshots)
3. The environment (e.g. operating system and version of manim)</p>
<h3>If this is a feature request:</h3>
<p>Include the motivation for making this change.</p>;1586189026000;1586189177000
609886736;595237885;yutharsan;<p>C:\Manim&gtSEMICOLONpython -m manim from_3b1b\old\basel\basel.py LightIndicator -pm
Traceback (most recent call last):
  File "C:\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "<strong>main</strong>", mod_spec)
  File "C:\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Manim\manim.py", line 5, in <module>
    manimlib.main()
  File "C:\Manim\manimlib__init__.py", line 11, in main
    config = manimlib.config.get_configuration(args)
  File "C:\Manim\manimlib\config.py", line 185, in get_configuration
    module = get_module(args.file)
  File "C:\Manim\manimlib\config.py", line 180, in get_module
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "from_3b1b\old\basel\basel.py", line 6, in <module>
    from once_useful_constructs.light import *
ModuleNotFoundError: No module named 'once_useful_constructs'</p>;1586189177000;1586189177000
595082889;595082889;GregoryMorse;<p><code>TEMPLATE_TEXT_FILE_BODY_TABULAR = TEMPLATE_TEXT_FILE_BODY.replace("\\usepackage[english]{babel}\n", "\\usepackage[english]{babel}\n\\usepackage{caption}\n\\usepackage{multicol}\n\\usepackage{multirow}\n")
tblBitwiseAnd = r"""
\begin{table}
    \centering
    \begin{tabular}{|c|l|c|l|c|}
         \hline
         \textbf{$a$} &ampSEMICOLON \multirow{5}{*}{$\&ampSEMICOLON$} &ampSEMICOLON \textbf{$b$} &ampSEMICOLON \multirow{5}{*}{$=$} &ampSEMICOLON \textbf{$a\&ampSEMICOLON b$} \\ \cline{1-1} \cline{3-3} \cline{5-5}
         0 &ampSEMICOLON &ampSEMICOLON 0 &ampSEMICOLON &ampSEMICOLON 0 \\ \cline{1-1} \cline{3-3} \cline{5-5}
         0 &ampSEMICOLON &ampSEMICOLON 1 &ampSEMICOLON &ampSEMICOLON 0 \\ \cline{1-1} \cline{3-3} \cline{5-5}
         1 &ampSEMICOLON &ampSEMICOLON 0 &ampSEMICOLON &ampSEMICOLON 0 \\ \cline{1-1} \cline{3-3} \cline{5-5}
         1 &ampSEMICOLON &ampSEMICOLON 1 &ampSEMICOLON &ampSEMICOLON 1 \\ \hline
    \end{tabular}
    \caption*{Bitwise And/Multiplication\\($a\&ampSEMICOLON b\equiv a*b$)}
\end{table}
"""
tbl = TextMobject(tblBitwiseAnd, template_tex_file_body=TEMPLATE_TEXT_FILE_BODY_TABULAR)</code></p>
<p>However since TextMobject defines CONFIG as does its base class TexMobject, and the further base class SingleStringTexMobject also defines it, TexMobject digest_config does not cause it to load into self.CONFIG and then when TexMobject calls break_up_by_substrings which uses self.CONFIG to initialize further SingleStringTexMobject objects, they will not use the correct template.  </p>
<p>It looks like any reference to self.CONFIG would be less than ideal because the object dictionary is updated based on a combination of things, and self.CONFIG is basically only supposed to be used in the constructor.</p>
<p>The fact that break_up_by_substrings references certainly looks like a bug.</p>
<p>Workaround:
```
      o = TextMobject.CONFIG.copy()
      o['template_tex_file_body']=TEMPLATE_TEXT_FILE_BODY_TABULAR
      tbl = TextMobject(tblBitwiseAnd, CONFIG=o, template_tex_file_body=TEMPLATE_TEXT_FILE_BODY_TABULAR)</p>
<p><code></code></p>;1586176871000;1586186957000
594781736;594781736;shagmwp;<p>Hi, I loved your Youtube video and thank you for your code. I updated the code to include Hospitalizations as well as Deceased.  This way you can see how scary a virus truly it. The common cold could have lots of infections, but no hospitalizations or deceased.  This way you can apply percentages of infected get hospitalized, and percentage of hospitalized pass away. With this you could add a "IF hospitals are over crowded death rate increase" condition (not added yet).</p>
<p>I also added an "at risk" attribute to model the percentage of population that are at higher risk to get hospitalized.</p>
<p>And lastly I added a healthy exposure simulation, where people not at risk can purposely expose themselves (think chicken pox parties) and then quarantine. You would be surprised.</p>
<p>I played with the models to increase infections time to 14 days, as well as increase time to symptomatic to 4 days, this would be more realistic  to Covid.</p>
<p>And lastly, I have one bug fix. Your last checking broke the multi city models.</p>
<p>Thanks for contributing to manim!</p>
<p><strong>Please ensure that your pull request works with the latest version of manim.</strong>
You should also include:</p>
<ol>
<li>The motivation for making this change (or link the relevant issues)</li>
<li>How you tested the new behavior (e.g. a minimal working example, before/after
screenshots, gifs, commands, etc.) This is rather informal at the moment, but
the goal is to show us how you know the pull request works as intended.</li>
</ol>;1586144763000;1586144763000
594344570;594344570;yutharsan;<h3>If this is a support request:</h3>
<p><strong>Please attempt to solve the problem on your own before opening an issue.</strong>
Between old issues, StackOverflow, and Google, you should be able to find
solutions to most of the common problems.</p>
<p>Include at least:
1. Steps to reproduce the issue (e.g. the command you ran)
2. The unexpected behavior that occurred (e.g. error messages or screenshots)
3. The environment (e.g. operating system and version of manim)</p>
<h3>If this is a feature request:</h3>
<p>Include the motivation for making this change.</p>;1586077811000;1586079820000
609388427;594344570;einarf;<p>You can manually upgrade each pacakge to the latest version. I'm guessing you are using python 3.8 and will need newver versions for it to work.</p>
<p>There are also some missing deps if you are in the shaders branch. moderngl-window, ipython and mapbox-earcut I think.</p>;1586079736000;1586079820000
592977009;592977009;lewis-cooper;<p>Is there anyway to take a graph thats already been plotted and find the gradient at each point of the graph and plot the values for the gradient on the graph?</p>;1585869816000;1585869816000
591643321;591643321;chenxijun;<p>To fix the bug that causes <code>Type Error: run() missing 1 required positional argument: 'self'</code></p>
<p>Thanks for contributing to manim!</p>
<p><strong>Please ensure that your pull request works with the latest version of manim.</strong>
You should also include:</p>
<ol>
<li>The motivation for making this change (or link the relevant issues)</li>
<li>How you tested the new behavior (e.g. a minimal working example, before/after
screenshots, gifs, commands, etc.) This is rather informal at the moment, but
the goal is to show us how you know the pull request works as intended.</li>
</ol>;1585719943000;1585895530000
591325292;591325292;applemonkey496;<p>I just wanted to point out that pycairo can be found in <a href="https://github.com/pygobject/pycairo">this GitHub repo</a>, and that there it says installing it is as simple as <code>pip install pycairo</code>. Am I missing something or should this information be included in the docs? It would be much easier than locating and installing the appropriate version from <a href="https://www.lfd.uci.edu/~gohlke/pythonlibs/#pycairo">here</a> as the docs say. </p>
<p>** Also, this is my first issue, so please tell me if I'm doing something wrong.</p>;1585679292000;1585690103000
606804397;591325292;einarf;<p>There is more to it:
https://pycairo.readthedocs.io/en/latest/getting_started.html</p>
<blockquote>
<p>Installing Pycairo requires pkg-config and cairo including its headers</p>
</blockquote>
<p>The pre-compiled versions LFD is providing is probably the easiest option on windows.</p>;1585680274000;1585680274000
606809639;591325292;applemonkey496;<p>@einarf, so does that mean that if I'm on MacOS (with Homebrew), to install pycairo, I would do <code>brew install cairo pkg-config</code> and then <code>pip3 install pycairo</code>? That would still be helpful to have that added to the docs, because right now I'd say the instructions for MacOS installation are quite vague.</p>;1585680871000;1585684511000
606886388;591325292;einarf;<p>Yeah they are. Could be good to PR doc changes in the pycairo project and link to that from Manim.</p>;1585690103000;1585690103000
590532375;590532375;badele;<p>Create some missing ressources for generation the all the scenes of "from_3b1b/old/spirals" </p>
<p>This can be used as a tutorials base :)</p>;1585596199000;1585596199000
590505279;590505279;lewis-cooper;<p>Is it possible to change the font that the axis values and the axis titles have?</p>;1585593463000;1585593463000
590498714;590498714;kolibril13;<p>Some GLSL compilers are more sensitive to types than others. With this fix, my Intel HD Graphics 5500 (Broadwell GT2) gives no error on this script anymore.</p>
<p>Thanks for contributing to manim!</p>
<p><strong>Please ensure that your pull request works with the latest version of manim.</strong>
You should also include:</p>
<ol>
<li>The motivation for making this change (or link the relevant issues)</li>
<li>How you tested the new behavior (e.g. a minimal working example, before/after
screenshots, gifs, commands, etc.) This is rather informal at the moment, but
the goal is to show us how you know the pull request works as intended.</li>
</ol>;1585592813000;1585706686000
606171037;590498714;einarf;<p>Input from a moderngl maintainer:</p>
<p>There might be more of this around in the glsl code. I think where floats should be expected it's typically a good idea to use <code>1.</code> or <code>1.0</code> instead of <code>1</code> as it could lead to strange behaviour with some drivers and in this case actually a compiler error.</p>
<p>This fix however makes Manim run at least!</p>
<p>EDIT: Just saw this is also fixed in #943</p>;1585593508000;1585607539000
606251721;590498714;zavden;<p>It works! thanks.</p>;1585602640000;1585602640000
590497153;590497153;azarzadavila;<p>I added some examples and explanations about the content of the file geometry.py and animation.
I tried to be exhaustive about the files I have inspected to write these modifications.
I hosted this branch on readthedocs with links <a href="https://azarzadavila-manim.readthedocs.io/en/latest/animation.html">animation</a>, <a href="https://azarzadavila-manim.readthedocs.io/en/latest/geometry.html">geometry</a>.</p>;1585592666000;1585592666000
590320691;590320691;lewis-cooper;<p>Trying to plot some data points using sublime text and I get this error. Any help would be appreciated.</p>
<p>Command ran:
```
from manimlib.imports import *</p>
<h1>Function to return data from .csv file to an array</h1>
<p>def get_coords_from_csv(coronavirus_data):
    import csv
    coords = []
    with open(f'coronavirus_data.csv', 'r') as csvFile:
        reader = csv.reader(csvFile)
        for row in reader:
            x,y = row
            coord = [float(x),float(y)]
            coords.append(coord)
    csvFile.close()
    return coords</p>
<p>class GraphFromData(GraphScene):
    def get_points_from_coords(self,coords):
        return[
        self.coords_to_point(px,py)
        for px,py in coords
        ]</p>
<pre><code>def get_dots_from_coords(self, coords, radius=0.1):
    points = self.get_points_from_coords(coords)
    dots = VGroup(*[
        Dot(radius=radius).move_to([px,py,pz])
        for px,py,pz in points
        ]
    )
    return dots
</code></pre>
<p>class DiscreteGraphFromSetPoints(VMobject):
    def <strong>init</strong>(self, set_of_points, <strong>kwargs):
        super().<strong>init</strong>(</strong>kwargs)
        self.set_points_as_corners(set_of_points)</p>
<p>class SmoothGraphFromSetPoints(VMobject):
    def <strong>init</strong>(self, set_of_points, <strong>kwargs):
        super().<strong>init</strong>(</strong>kwargs)
        self.set_points_smoothly(set_of_points)</p>
<h1>Graph with set of points</h1>
<p>class CustomGraph1(GraphFromData):
    def construct(self):
        self.setup_axes()
        coords = get_coords_from_csv("coronavirus_data")
        dots = self.get_dots_from_coords(coords)
        self.add(dots)
```</p>
<p>The error:
Traceback (most recent call last):
  File "/Users/lewis/Documents/animation/manim/manimlib/extract_scene.py", line 155, in main
    scene = SceneClass(**scene_kwargs)
  File "/Users/lewis/Documents/animation/manim/manimlib/scene/scene.py", line 53, in <strong>init</strong>
    self.construct()
  File "coronavirus.py", line 49, in construct
    coords = get_coords_from_csv("coronavirus_data")
  File "coronavirus.py", line 13, in get_coords_from_csv
    coord = [float(x),float(y)]
ValueError: could not convert string to float: '\ufeff1'</p>
<p>Many thanks :)</p>;1585577246000;1585577283000
590102894;590102894;Micoael;<p>I was trying to render a Linear transform scene to have a try at the animations.
When I enter the from3b1b folder and eola series and began rendering files, when the following errors are shown:</p>
<p>```
PS D:\3b1b\manim&gtSEMICOLON  python -m manim from_3b1b/old/eola/chapter3.py SimpleLinearTransformationScene -ps
Media will be written to D:\3b1b\manim\media. You can change this behavior with the --media_dir flag.</p>
<p>Traceback (most recent call last):
  File "D:\3b1b\manim\manimlib\extract_scene.py", line 155, in main
    scene = SceneClass(<strong>scene_kwargs)
  File "D:\3b1b\manim\manimlib\scene\scene.py", line 51, in <strong>init</strong>
    self.setup()
  File "D:\3b1b\manim\manimlib\scene\vector_space_scene.py", line 324, in setup
    </strong>self.background_plane_kwargs
  File "D:\3b1b\manim\manimlib\mobject\coordinate_systems.py", line 287, in <strong>init</strong>
    super.<strong>init</strong>(**kwargs)
TypeError: descriptor '<strong>init</strong>' of 'super' object needs an argument
```
Lots of thanks!</p>;1585557770000;1585706050000
589883404;589883404;timalive;<p>Added a "Death" state to the SIR model, that depends on a recovery/death probability.  I have also modelled a dependency on healthcare capacity. I would appreciate permission to put a video together describing the changes made and to publish it as a supplement to your excellent SIR model video.</p>
<p>I havent yet tested it on all the scenes, but have added it to a few key scenes. </p>
<p>Thanks for contributing to manim!</p>
<p><strong>Please ensure that your pull request works with the latest version of manim.</strong>
You should also include:</p>
<ol>
<li>The motivation for making this change (or link the relevant issues)</li>
<li>How you tested the new behavior (e.g. a minimal working example, before/after
screenshots, gifs, commands, etc.) This is rather informal at the moment, but
the goal is to show us how you know the pull request works as intended.</li>
</ol>;1585515300000;1585517129000
589661219;589661219;Lalourche;<p>Hi,
This is my first attempt at a pull request for manim.</p>
<p>It fixes some runtime errors when generating counting.py (no issue found on the subject).</p>
<p>```
    for point in self.get_template_configuration(place)
TypeError: get_template_configuration() takes 1 positional argument but 2 were given</p>
<pre><code>it.cycle(new_template)
</code></pre>
<p>NameError: name 'it' is not defined
```</p>
<p>These errors do not appear anymore after fix.
<a href="https://travis-ci.org/Lalourche/manim"><img alt="Build Status" src="https://travis-ci.org/Lalourche/manim.svg?branch=fix-counting" /></a></p>
<p>Cheers
Lalourche</p>;1585430123000;1585430123000
589568867;589568867;edasubert;<p><code>fragment_shader
===============
0:179(24): error: `return' with wrong type int, in function `sdf' returning float</code></p>
<p>Thanks for contributing to manim!</p>
<p><strong>Please ensure that your pull request works with the latest version of manim.</strong>
You should also include:</p>
<ol>
<li>The motivation for making this change (or link the relevant issues)</li>
<li>How you tested the new behavior (e.g. a minimal working example, before/after
screenshots, gifs, commands, etc.) This is rather informal at the moment, but
the goal is to show us how you know the pull request works as intended.</li>
</ol>;1585398815000;1585398815000
589400887;589400887;mvfs314;<p>I'd like to run the code:
<code>numeros4 = TexMobject(
            "\\mathbb{N}",
            "=",
            "\\{",
            "1",
            ",",
            "2",
            ",",
            "3",
            ",",
            "\\ldots",
            "\\}"
        )</code>
cause I wish I could change the color of the braces separately. But I got this error:
<code>Exception: Latex error converting to dvi.</code>
I've already tried using the command <code>Brace()</code> or the ideas found at reddit, but it didn't work at all. Is there a way to make it possible?</p>
<p>Thanks!</p>;1585341539000;1585400402000
605434438;589400887;Aathish04;<p>Have you installed LaTeX? You also need quite a few LaTeX packages with it as well.
Please post the full traceback and we may be able to help you better.</p>;1585395192000;1585395192000
605440295;589400887;mvfs314;<p>I certainly have. This is the basic LaTeX code, I just wanna write the braces separately. </p>
<p>Here the traceback:
<code>Traceback (most recent call last):
  File "C:\Manim\manim-master\manimlib\extract_scene.py", line 155, in main
    scene = SceneClass(**scene_kwargs)
  File "C:\Manim\manim-master\manimlib\scene\scene.py", line 53, in __init__
    self.construct()
  File "1.py", line 170, in construct
    self.numeros_naturais()
  File "1.py", line 241, in numeros_naturais
    "\\}"
  File "C:\Manim\manim-master\manimlib\mobject\svg\tex_mobject.py", line 153, in __init__
    self.break_up_by_substrings()
  File "C:\Manim\manim-master\manimlib\mobject\svg\tex_mobject.py", line 184, in break_up_by_substrings
    sub_tex_mob = SingleStringTexMobject(tex_string, **config)
  File "C:\Manim\manim-master\manimlib\mobject\svg\tex_mobject.py", line 44, in __init__
    self.template_tex_file_body
  File "C:\Manim\manim-master\manimlib\utils\tex_file_writing.py", line 19, in tex_to_svg_file
    dvi_file = tex_to_dvi(tex_file)
  File "C:\Manim\manim-master\manimlib\utils\tex_file_writing.py", line 67, in tex_to_dvi
    "See log output above or the log file: %s" % log_file)
Exception: Latex error converting to dvi. See log output above or the log file: ./media\Tex\0447ed74899f41fb.log</code></p>
<p>Thanks!</p>;1585398194000;1585398194000
605440741;589400887;Aathish04;<p>Hmm, the traceback points to a log file in <code>./media\Tex\0447ed74899f41fb.log</code>. Could you please post the contents of that as well?</p>
<p>Also, I've heard people saying that it isn't possible to write a single brace in LaTeX. You use Brace() for that, but you say you have tried that as well.</p>
<p>In any case, the contents of the log file would be mighty helpful.</p>
<blockquote>
<p>On 28-Mar-2020, at 17:53, mvfs314 <a href="&#109SEMICOLON&#97SEMICOLON&#105SEMICOLON&#108SEMICOLON&#116SEMICOLON&#111SEMICOLON&#58SEMICOLON&#110SEMICOLON&#111SEMICOLON&#116SEMICOLON&#105SEMICOLON&#102SEMICOLON&#105SEMICOLON&#99SEMICOLON&#97SEMICOLON&#116SEMICOLON&#105SEMICOLON&#111SEMICOLON&#110SEMICOLON&#115SEMICOLON&#64SEMICOLON&#103SEMICOLON&#105SEMICOLON&#116SEMICOLON&#104SEMICOLON&#117SEMICOLON&#98SEMICOLON&#46SEMICOLON&#99SEMICOLON&#111SEMICOLON&#109SEMICOLON">&#110SEMICOLON&#111SEMICOLON&#116SEMICOLON&#105SEMICOLON&#102SEMICOLON&#105SEMICOLON&#99SEMICOLON&#97SEMICOLON&#116SEMICOLON&#105SEMICOLON&#111SEMICOLON&#110SEMICOLON&#115SEMICOLON&#64SEMICOLON&#103SEMICOLON&#105SEMICOLON&#116SEMICOLON&#104SEMICOLON&#117SEMICOLON&#98SEMICOLON&#46SEMICOLON&#99SEMICOLON&#111SEMICOLON&#109SEMICOLON</a> wrote:</p>
<p>I certainly have. This is the basic LaTeX code, I just wanna write the braces separately.</p>
<p>Here the traceback:
Traceback (most recent call last): File "C:\Manim\manim-master\manimlib\extract_scene.py", line 155, in main scene = SceneClass(<strong>scene_kwargs) File "C:\Manim\manim-master\manimlib\scene\scene.py", line 53, in <strong>init</strong> self.construct() File "1.py", line 170, in construct self.numeros_naturais() File "1.py", line 241, in numeros_naturais "\}" File "C:\Manim\manim-master\manimlib\mobject\svg\tex_mobject.py", line 153, in <strong>init</strong> self.break_up_by_substrings() File "C:\Manim\manim-master\manimlib\mobject\svg\tex_mobject.py", line 184, in break_up_by_substrings sub_tex_mob = SingleStringTexMobject(tex_string, </strong>config) File "C:\Manim\manim-master\manimlib\mobject\svg\tex_mobject.py", line 44, in <strong>init</strong> self.template_tex_file_body File "C:\Manim\manim-master\manimlib\utils\tex_file_writing.py", line 19, in tex_to_svg_file dvi_file = tex_to_dvi(tex_file) File "C:\Manim\manim-master\manimlib\utils\tex_file_writing.py", line 67, in tex_to_dvi "See log output above or the log file: %s" % log_file) Exception: Latex error converting to dvi. See log output above or the log file: ./media\Tex\0447ed74899f41fb.log</p>
<p>Thanks!</p>
<p>—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub <a href="https://github.com/3b1b/manim/issues/941#issuecomment-605440295">https://github.com/3b1b/manim/issues/941#issuecomment-605440295</a>, or unsubscribe <a href="https://github.com/notifications/unsubscribe-auth/AH5H6JBA2UIU6JM4LV7UW6TRJXT35ANCNFSM4LVIZ2FA">https://github.com/notifications/unsubscribe-auth/AH5H6JBA2UIU6JM4LV7UW6TRJXT35ANCNFSM4LVIZ2FA</a>.
</p>
</blockquote>;1585398468000;1585398468000
605441286;589400887;mvfs314;<p>Here all the log file:
`This is pdfTeX, Version 3.14159265-2.6-1.40.21 (MiKTeX 2.9.7300 64-bit) (preloaded format=latex 2020.2.25)  28 MAR 2020 09:20
entering extended mode
**./media/Tex/0447ed74899f41fb.tex
(media/Tex/0447ed74899f41fb.tex
LaTeX2e &ltSEMICOLON2020-02-02&gtSEMICOLON patch level 3
L3 programming layer &ltSEMICOLON2020-02-21&gtSEMICOLON
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/standalone\standal
one.cls"
Document Class: standalone 2018/03/26 v1.3a Class to compile TeX sub-files stan
dalone
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/tools\shellesc.sty
"
Package: shellesc 2019/11/08 v1.0c unified shell escape interface for LaTeX
Package shellesc Info: Restricted shell escape enabled on input line 77.
)
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/iftex\ifluatex.s
ty"
Package: ifluatex 2019/10/25 v1.5 ifluatex legacy package. Use iftex instead.
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/iftex\iftex.sty"
Package: iftex 2019/11/07 v1.0c TeX engine tests
))
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/xkeyval\xkeyval.st
y"
Package: xkeyval 2014/12/03 v2.7a package option processing (HA)</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/xkeyval\xkeyval.
tex"
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/xkeyval\xkvutils
.tex"
\XKV@toks=\toks14
\XKV@tempa@toks=\toks15</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/xkeyval\keyval.t
ex"))
\XKV@depth=\count167
File: xkeyval.tex 2014/12/03 v2.7a key=value parser (HA)
))
\sa@internal=\count168
\c@sapage=\count169</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/standalone\standal
one.cfg"
File: standalone.cfg 2018/03/26 v1.3a Default configuration file for 'standalon
e' class
) ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/base\article.cls
"
Document Class: article 2019/12/20 v1.4l Standard LaTeX document class
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/base\size10.clo"
File: size10.clo 2019/12/20 v1.4l Standard LaTeX file (size option)
)
\c@part=\count170
\c@section=\count171
\c@subsection=\count172
\c@subsubsection=\count173
\c@paragraph=\count174
\c@subparagraph=\count175
\c@figure=\count176
\c@table=\count177
\abovecaptionskip=\skip47
\belowcaptionskip=\skip48
\bibindent=\dimen134
) ("C:\Program Files\MiKTeX 2.9\tex/latex\preview\preview.sty"
Package: preview 2017/04/24 11.91 (AUCTeX/preview-latex)</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/luatex85\luatex8
5.sty"
Package: luatex85 2016/06/15 v1.4 pdftex aliases for luatex
) ("C:\Program Files\MiKTeX 2.9\tex/latex\preview\prtightpage.def"
\PreviewBorder=\dimen135
)
\pr@snippet=\count178
\pr@box=\box45
\pr@output=\toks16
)) ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/babel\babel.s
ty"
Package: babel 2020/02/14 3.40 The Babel package
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/babel\switch.def
"
File: switch.def 2020/02/14 3.40 Babel switching mechanism
)</p>
<hr />
<ul>
<li>Local config file bblopts.cfg used
*
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/arabi\bblopts.cfg"
File: bblopts.cfg 2005/09/08 v0.1 add Arabic and Farsi to "declared" options of
 babel
)
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/babel-english\engl
ish.ldf"
Language: english 2017/06/06 v3.3r English support from the babel system
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/babel\babel.def"
File: babel.def 2020/02/14 3.40 Babel common definitions
\babel@savecnt=\count179
\U@D=\dimen136</li>
</ul>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/generic/babel\txtbabel.d
ef")
\bbl@readstream=\read2
\bbl@dirlevel=\count180
)
Package babel Info: \l@canadian = using hyphenrules for english
(babel)             (\language0) on input line 102.
Package babel Info: \l@australian = using hyphenrules for ukenglish
(babel)             (\language77) on input line 105.
Package babel Info: \l@newzealand = using hyphenrules for ukenglish
(babel)             (\language77) on input line 108.
)) ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/base\inputenc.s
ty"
Package: inputenc 2018/08/11 v1.3c Input encoding file
\inpenc@prehook=\toks17
\inpenc@posthook=\toks18
) ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/base\fontenc.sty
"
Package: fontenc 2020/02/11 v2.0o Standard LaTeX package
)
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsmath\amsmath.st
y"
Package: amsmath 2020/01/20 v2.17e AMS math features
\@mathmargin=\skip49
For additional information on amsmath, use the `?' option.</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsmath\amstext.st
y"
Package: amstext 2000/06/29 v2.01 AMS text
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsmath\amsgen.sty
"
File: amsgen.sty 1999/11/30 v2.0 generic functions
\@emptytoks=\toks19
\ex@=\dimen137
)) ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsmath\amsbsy.
sty"
Package: amsbsy 1999/11/29 v1.2d Bold Symbols
\pmbraise@=\dimen138
) ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsmath\amsopn.s
ty"
Package: amsopn 2016/03/08 v2.02 operator names
)
\inf@bad=\count181
LaTeX Info: Redefining \frac on input line 227.
\uproot@=\count182
\leftroot@=\count183
LaTeX Info: Redefining \overline on input line 389.
\classnum@=\count184
\DOTSCASE@=\count185
LaTeX Info: Redefining \ldots on input line 486.
LaTeX Info: Redefining \dots on input line 489.
LaTeX Info: Redefining \cdots on input line 610.
\Mathstrutbox@=\box46
\strutbox@=\box47
\big@size=\dimen139
LaTeX Font Info:    Redeclaring font encoding OML on input line 733.
LaTeX Font Info:    Redeclaring font encoding OMS on input line 734.
\macc@depth=\count186
\c@MaxMatrixCols=\count187
\dotsspace@=\muskip16
\c@parentequation=\count188
\dspbrk@lvl=\count189
\tag@help=\toks20
\row@=\count190
\column@=\count191
\maxfields@=\count192
\andhelp@=\toks21
\eqnshift@=\dimen140
\alignsep@=\dimen141
\tagshift@=\dimen142
\tagwidth@=\dimen143
\totwidth@=\dimen144
\lineht@=\dimen145
\@envbody=\toks22
\multlinegap=\skip50
\multlinetaggap=\skip51
\mathdisplay@stack=\toks23
LaTeX Info: Redefining [ on input line 2859.
LaTeX Info: Redefining ] on input line 2860.
)
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsfonts\amssymb.s
ty"
Package: amssymb 2013/01/14 v3.01 AMS font symbols</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsfonts\amsfonts.
sty"
Package: amsfonts 2013/01/14 v3.01 Basic AMSFonts support
\symAMSa=\mathgroup4
\symAMSb=\mathgroup5
LaTeX Font Info:    Redeclaring math symbol \hbar on input line 98.
LaTeX Font Info:    Overwriting math alphabet <code>\mathfrak' in version</code>bold'
(Font)                  U/euf/m/n --&gtSEMICOLON U/euf/b/n on input line 106.
)) ("C:\Program Files\MiKTeX 2.9\tex/latex\doublestroke\dsfont.sty"
Package: dsfont 1995/08/01 v0.1 Double stroke roman fonts
) ("C:\Program Files\MiKTeX 2.9\tex/latex\setspace\setspace.sty"
Package: setspace 2011/12/19 v6.7a set line spacing
) ("C:\Program Files\MiKTeX 2.9\tex/latex\tipa\tipa.sty"
Package: tipa 2002/08/08 TIPA version 1.1
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/base\fontenc.sty"
Package: fontenc 2020/02/11 v2.0o Standard LaTeX package
("C:\Program Files\MiKTeX 2.9\tex/latex\tipa\t3enc.def"
File: t3enc.def 2001/12/31 T3 encoding
Now handling font encoding T3 ...
... no UTF-8 mapping file for font encoding T3
LaTeX Font Info:    Trying to load font information for T1+cmss on input line 3
57.
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/base\t1cmss.fd"
File: t1cmss.fd 2019/12/16 v2.5j Standard LaTeX font definitions
)))) ("C:\Program Files\MiKTeX 2.9\tex/latex\relsize\relsize.sty"
Package: relsize 2013/03/29 ver 4.1
) ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/base\textcomp.st
y"
Package: textcomp 2020/02/02 v2.0n Standard LaTeX package
) ("C:\Program Files\MiKTeX 2.9\tex/latex\jknappen\mathrsfs.sty"
Package: mathrsfs 1996/01/01 Math RSFS package v1.0 (jk)
\symrsfs=\mathgroup6
) ("C:\Program Files\MiKTeX 2.9\tex/latex\fundus-calligra\calligra.sty"
Package: calligra 2012/04/10 v1.9 LaTeX package calligra
) ("C:\Program Files\MiKTeX 2.9\tex/latex\wasysym\wasysym.sty"
Package: wasysym 2020/01/19 v2.4 Wasy-2 symbol support package
\symwasy=\mathgroup7
LaTeX Font Info:    Overwriting symbol font <code>wasy' in version</code>bold'
(Font)                  U/wasy/m/n --&gtSEMICOLON U/wasy/b/n on input line 93.
)
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/ragged2e\ragged2e.
sty"
Package: ragged2e 2019/07/28 v2.2 ragged2e Package (MS)
("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/ms\everysel.sty"
Package: everysel 2011/10/28 v1.2 EverySelectfont Package (MS)
)
\CenteringLeftskip=\skip52
\RaggedLeftLeftskip=\skip53
\RaggedRightLeftskip=\skip54
\CenteringRightskip=\skip55
\RaggedLeftRightskip=\skip56
\RaggedRightRightskip=\skip57
\CenteringParfillskip=\skip58
\RaggedLeftParfillskip=\skip59
\RaggedRightParfillskip=\skip60
\JustifyingParfillskip=\skip61
\CenteringParindent=\skip62
\RaggedLeftParindent=\skip63
\RaggedRightParindent=\skip64
\JustifyingParindent=\skip65
) ("C:\Program Files\MiKTeX 2.9\tex/latex\physics\physics.sty"
Package: physics </p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/l3packages/xparse\
xparse.sty" ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/l3kern
el\expl3.sty"
Package: expl3 2020-02-21 L3 programming layer (loader) </p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/l3backend\l3backen
d-dvips.def"
File: l3backend-dvips.def 2020-02-23 L3 backend support: dvips
\l__pdf_internal_box=\box48
\g__pdf_backend_object_int=\count193
\l__pdf_backend_content_box=\box49
\l__pdf_backend_model_box=\box50
\g__pdf_backend_annotation_int=\count194
\g__pdf_backend_link_int=\count195
\g__pdf_backend_link_sf_int=\count196
))
Package: xparse 2020-02-14 L3 Experimental document command parser
\l__xparse_current_arg_int=\count197
\g__xparse_grabber_int=\count198
\l__xparse_m_args_int=\count199
\l__xparse_v_nesting_int=\count266
)) ("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/xcolor\xcolor.s
ty"
Package: xcolor 2016/05/11 v2.12 LaTeX color extensions (UK)</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/graphics-cfg\color
.cfg"
File: color.cfg 2016/01/02 v1.6 sample color configuration
)
Package xcolor Info: Driver file: dvips.def on input line 225.</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/graphics-def\dvips
.def"
File: dvips.def 2017/06/20 v3.1d Graphics/color driver for dvips
)
Package xcolor Info: Model <code>cmy' substituted by</code>cmy0' on input line 1348.
Package xcolor Info: Model <code>RGB' extended on input line 1364.
Package xcolor Info: Model</code>HTML' substituted by <code>rgb' on input line 1366.
Package xcolor Info: Model</code>Hsb' substituted by <code>hsb' on input line 1367.
Package xcolor Info: Model</code>tHsb' substituted by <code>hsb' on input line 1368.
Package xcolor Info: Model</code>HSB' substituted by <code>hsb' on input line 1369.
Package xcolor Info: Model</code>Gray' substituted by <code>gray' on input line 1370.
Package xcolor Info: Model</code>wave' substituted by `hsb' on input line 1371.
) ("C:\Program Files\MiKTeX 2.9\tex/latex\microtype\microtype.sty"
Package: microtype 2019/11/18 v2.7d Micro-typographical refinements (RS)
\MT@toks=\toks24
\MT@count=\count267
LaTeX Info: Redefining \textls on input line 790.
\MT@outer@kern=\dimen146
LaTeX Info: Redefining \textmicrotypecontext on input line 1354.
\MT@listname@count=\count268
("C:\Program Files\MiKTeX 2.9\tex/latex\microtype\microtype-pdftex.def"
File: microtype-pdftex.def 2019/11/18 v2.7d Definitions specific to pdftex (RS)</p>
<p>LaTeX Info: Redefining \lsstyle on input line 914.
LaTeX Info: Redefining \lslig on input line 914.
\MT@outer@space=\skip66
)
Package microtype Info: Loading configuration file microtype.cfg.
("C:\Program Files\MiKTeX 2.9\tex/latex\microtype\microtype.cfg"
File: microtype.cfg 2019/11/18 v2.7d microtype main configuration file (RS)
)) (C:\Manim\manim-master\media\Tex\0447ed74899f41fb.aux)
\openout1 = `0447ed74899f41fb.aux'.</p>
<p>LaTeX Font Info:    Checking defaults for OML/cmm/m/it on input line 24.
LaTeX Font Info:    ... okay on input line 24.
LaTeX Font Info:    Checking defaults for OMS/cmsy/m/n on input line 24.
LaTeX Font Info:    ... okay on input line 24.
LaTeX Font Info:    Checking defaults for OT1/cmr/m/n on input line 24.
LaTeX Font Info:    ... okay on input line 24.
LaTeX Font Info:    Checking defaults for T1/cmr/m/n on input line 24.
LaTeX Font Info:    ... okay on input line 24.
LaTeX Font Info:    Checking defaults for TS1/cmr/m/n on input line 24.
LaTeX Font Info:    ... okay on input line 24.
LaTeX Font Info:    Checking defaults for OMX/cmex/m/n on input line 24.
LaTeX Font Info:    ... okay on input line 24.
LaTeX Font Info:    Checking defaults for U/cmr/m/n on input line 24.
LaTeX Font Info:    ... okay on input line 24.
LaTeX Font Info:    Checking defaults for T3/cmr/m/n on input line 24.
LaTeX Font Info:    Trying to load font information for T3+cmr on input line 24
.
("C:\Program Files\MiKTeX 2.9\tex/latex\tipa\t3cmr.fd"
File: t3cmr.fd 2001/12/31 TIPA font definitions
)
LaTeX Font Info:    ... okay on input line 24.
Preview: Fontsize 10pt
ABD: EverySelectfont initializing macros
LaTeX Info: Redefining \selectfont on input line 24.
LaTeX Info: Redefining \microtypecontext on input line 24.
Package microtype Info: Generating DVI output.
Package microtype Info: Character protrusion enabled (level 2).
Package microtype Info: Using default protrusion set `alltext'.
Package microtype Info: No font expansion.
Package microtype Info: No adjustment of tracking.
Package microtype Info: No adjustment of interword spacing.
Package microtype Info: No adjustment of character kerning.
("C:\Program Files\MiKTeX 2.9\tex/latex\microtype\mt-cmr.cfg"
File: mt-cmr.cfg 2013/05/19 v2.2 microtype config. file: Computer Modern Roman 
(RS)
)
LaTeX Font Info:    Trying to load font information for U+msa on input line 26.</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsfonts\umsa.fd"
File: umsa.fd 2013/01/14 v3.01 AMS symbols A
) ("C:\Program Files\MiKTeX 2.9\tex/latex\microtype\mt-msa.cfg"
File: mt-msa.cfg 2006/02/04 v1.1 microtype config. file: AMS symbols (a) (RS)
)
LaTeX Font Info:    Trying to load font information for U+msb on input line 26.</p>
<p>("C:\Users\mvfs3\AppData\Local\Programs\MiKTeX 2.9\tex/latex/amsfonts\umsb.fd"
File: umsb.fd 2013/01/14 v3.01 AMS symbols B
) ("C:\Program Files\MiKTeX 2.9\tex/latex\microtype\mt-msb.cfg"
File: mt-msb.cfg 2005/06/01 v1.0 microtype config. file: AMS symbols (b) (RS)
)
LaTeX Font Info:    Trying to load font information for U+rsfs on input line 26
.
("C:\Program Files\MiKTeX 2.9\tex/latex\jknappen\ursfs.fd"
File: ursfs.fd 1998/03/24 rsfs font definition file (jk)
)
LaTeX Font Info:    Trying to load font information for U+wasy on input line 26
.
("C:\Program Files\MiKTeX 2.9\tex/latex\wasysym\uwasy.fd"
File: uwasy.fd 2020/01/19 v2.4 Wasy-2 symbol font definitions
)
! Argument of \align* has an extra }.
<inserted text> 
                \par 
l.27 {}</p>
<p>Here is how much of TeX's memory you used:
 7327 strings out of 481508
 111190 string characters out of 2912081
 392778 words of memory out of 3000000
 22471 multiletter control sequences out of 15000+200000
 538995 words of font info for 43 fonts, out of 3000000 for 9000
 1141 hyphenation exceptions out of 8191
 43i,3n,60p,354b,212s stack positions out of 5000i,500n,10000p,200000b,50000s</p>
<p>No pages of output.
`</p>;1585398777000;1585398777000
605441345;589400887;mvfs314;<p>And, about the <code>Braces()</code> it's not very useful when I want to use in a single text line :/</p>;1585398807000;1585398807000
605444417;589400887;Aathish04;<p>This appears to be a LaTeX specific problem. </p>
<p>You simply cannot use a single brace separately in LaTeX, and the way manim renders LaTeX (it uses \align a lot), the parser keeps thinking it is seeing a single brace.</p>
<p>For the time being, i suggest you use two Brace()es, which is tedious, but perhaps someone more experienced with LaTeX can help you out.</p>
<p>If you feel like probing the problem deeper, you can modify tex_template.tex, and see if you can get it to work.</p>
<p>All the best SEMICOLON)</p>;1585400402000;1585400402000
589007446;589007446;nosnakeob;<p>When I use:
```
t = TextMobject('aaaaa', '''a</p>
<p>a''', 'aaaaa')
```
<img alt="批注 2020-03-27 173840" src="https://user-images.githubusercontent.com/47769817/77742690-d5367800-7051-11ea-8324-bb71627aa300.png" /></p>
<p>but when I use:
<code>t = TextMobject('aaaaaaaaa', '\n\n', 'aaaaaaa')</code>
or:
```
t = TextMobject('aaaaa', '''</p>
<pre><code>     ''', 'aaaaa')
</code></pre>
<p>```
<img alt="批注 2020-03-27 173932" src="https://user-images.githubusercontent.com/47769817/77742786-fdbe7200-7051-11ea-8470-cbc7933e64e9.png" /></p>
<p>It's not what I want. Are there other ways to add a new line when using <code>TextMobject</code> like arrays?</p>;1585302056000;1585304979000
604925408;589007446;qo4on;<p><code>'\n'.join(['First_line', 'Second_line', 'Third_line'])
TextMobject('First_line\nSecond_line\nThird_line')</code></p>;1585304793000;1585304979000
588937811;588937811;Ranoiaetep;<p>Removed pycairo section on Directly (Windows), as its included in requirements.txt
Removed changing pycairo version number in the same section, version number included in requirements.txt is &gtSEMICOLON= 1.18.1</p>;1585293672000;1585293811000
588714915;588714915;MatejVe;<p>I have succesfully installed all the packages required for working in manim. Problem arrises when trying to run the example_scenes.py from the console. Console reports no errors and it seems like everything went smoothly, however no video windows pop up and actually show the animation. I am working on windows 10, python 3.7.3 32bit</p>;1585255317000;1585256023000
604680067;588714915;MatejVe;<p>Figured it out. Intead of doing 
python3 manim.py example_scenes.py SquareToCircle -pl
in the console, I have done
manim example_scenes.py SquareToCircle -pl
and it works like it should now.
Still not sure why the first thing didn't work though</p>;1585256023000;1585256023000
588606969;588606969;qo4on;<p>I get an error in Colab trying to run Manim from "shaders" branch. I see <code>tqdm</code> for each animation, that means they are successful. But then it gets into a loop and I have to stop it with the error:
<code>Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/content/manim/manim.py", line 5, in &ltSEMICOLONmodule&gtSEMICOLON
    manimlib.main()
  File "/content/manim/manimlib/__init__.py", line 14, in main
    scene.run()
  File "/content/manim/manimlib/scene/scene.py", line 77, in run
    self.tear_down()
  File "/content/manim/manimlib/scene/scene.py", line 96, in tear_down
    self.interact()
  File "/content/manim/manimlib/scene/scene.py", line 104, in interact
    self.update_frame()
  File "/content/manim/manimlib/scene/scene.py", line 139, in update_frame
    self.camera.clear()
  File "/content/manim/manimlib/camera/camera.py", line 114, in clear
    rgba = (*Color(self.background_color).get_rgb(), self.background_opacity)
  File "/usr/local/lib/python3.6/dist-packages/colour.py", line 988, in __init__
    self.web = color if color else 'black'
  File "/usr/local/lib/python3.6/dist-packages/colour.py", line 1006, in __setattr__
    fc(value)
  File "/usr/local/lib/python3.6/dist-packages/colour.py", line 1084, in set_web
    self.hex = web2hex(value)
  File "/usr/local/lib/python3.6/dist-packages/colour.py", line 1005, in __setattr__
    fc = getattr(self, 'set_' + label)
KeyboardInterrupt</code>
<strong>UPDATE:
If you want to try it in Colab, <a href="https://github.com/moderngl/moderngl/issues/392#issuecomment-605913255">the code is here</a>.</strong></p>;1585244760000;1585593681000
605602579;588606969;timalive;<p>I am getting the same issue when running in docker. Something to do with the fact that pyglet needs a display to run. There may be a solution using Xvfb</p>;1585470217000;1585470217000
605608314;588606969;qo4on;<p>xvfb doesn't work either. We need to change manim <code>window.py</code> backend from <code>pyglet</code> to <code>headless</code>.
<code>from moderngl_window.context.headless.window import Window as HeadlessWindow</code>
But I didn't managed to make it work.</p>;1585473755000;1585473797000
605677632;588606969;t-wolfeadam;<p>Running Win10, I couldn't get it to work when using the preview setting (-p) due to no configs for the window:
<code>Traceback (most recent call last):
  File "C:\Users\Tyler Wolfe-Adam\Documents\Python\manim-shaders\manim.py", line 5, in &ltSEMICOLONmodule&gtSEMICOLON
    manimlib.main()
  File "C:\Users\Tyler Wolfe-Adam\Documents\Python\manim-shaders\manimlib\__init__.py", line 11, in main
    scenes = manimlib.extract_scene.main(config)
  File "C:\Users\Tyler Wolfe-Adam\Documents\Python\manim-shaders\manimlib\extract_scene.py", line 104, in main
    scenes = get_scenes_to_render(all_scene_classes, config)
  File "C:\Users\Tyler Wolfe-Adam\Documents\Python\manim-shaders\manimlib\extract_scene.py", line 72, in get_scenes_to_render
    scene = scene_class(**scene_kwargs)
  File "C:\Users\Tyler Wolfe-Adam\Documents\Python\manim-shaders\manimlib\scene\scene.py", line 44, in __init__
    self.window = Window(self, **self.window_config)
  File "C:\Users\Tyler Wolfe-Adam\Documents\Python\manim-shaders\manimlib\window.py", line 21, in __init__
    super().__init__(**kwargs)
  File "C:\Program Files\Python37\lib\site-packages\moderngl_window\context\pyglet\window.py", line 54, in __init__
    config=config,
  File "C:\Program Files\Python37\lib\site-packages\pyglet\window\win32\__init__.py", line 130, in __init__
    super(Win32Window, self).__init__(*args, **kwargs)
  File "C:\Program Files\Python37\lib\site-packages\pyglet\window\__init__.py", line 591, in __init__
    config = screen.get_best_config(config)
  File "C:\Program Files\Python37\lib\site-packages\pyglet\canvas\base.py", line 194, in get_best_config
    raise window.NoSuchConfigException()
pyglet.window.NoSuchConfigException</code>
It <em>does</em> work when just writing to a file (-w). However, just for the shape examples. Anything with text doesn't work due to (unrelated) issues with dvisvgm not converting the dvi to svg.</p>;1585505829000;1585505829000
605678964;588606969;qo4on;<p>What text doesn't work? <code>tex_mobject</code> or <code>text_mobject</code>?</p>;1585506426000;1585506426000
605719004;588606969;t-wolfeadam;<blockquote>
<p>What text doesn't work? <code>tex_mobject</code> or <code>text_mobject</code>?</p>
</blockquote>
<p>Both, unfortunatelySEMICOLON I also tried modifying the example scenes with just using one or the other. Both result in <code>OSError: No file matching ./media\Tex\ab47de5016aed0d4.svg in image directory</code>.</p>
<p>I dug into it and it stems from the <code>dvisvgm</code> command (called from the <code>dvi_to_svg</code> function inside <code>\manimlib\utils\tex_file_writing.py</code>) - the command executes successfully but nothing gets outputted (no files or messages, even with the highest level of verbosity). I even tried searching my entire file system for the file and still nothing. It's really weird. </p>
<p>I'm currently getting it setup to try on WSL. Hopefully that works better. No luck on googling the problem either. I'll probably open a new thread somewhere if it doesn't work with WSL :/</p>
<p>Edit: The problem with dvisvgm seems to be a Windows thing. I found out it's returning exit code -1073740791, which is apparently a stack buffer overflow. dvisvgm works fine on WSL, so for now I've implemented a hacky solution where Windows waits until Ubuntu has converted the necessary files. </p>
<p>Edit 2: <strong>Found the fix!</strong> For future readers: Running <code>dvisvgm --version</code> reported 2.7. I ran <code>where dvisvgm</code> and it reported two versions: the v2.7 (installed in Program Files) showed before the other, which was part of MikTex and version 2.8.2. Removing the v2.7 copy fixed this issue. 
I am now able to render videos in this branch, as long as I'm writing to file (<code>w</code> flag) and not showing preview (<code>p</code> flag). </p>;1585524740000;1585588715000
605728064;588606969;t-wolfeadam;<p>Update: Ran into the same problems with the display. Tried using xvfb like so:
<code>xvfb-run -s "-screen 0 640x360x24" python3.7 -m manim example_scenes.py SquareToCircle -ql</code>
That returned the same error for me as when I tried to use the <code>q</code> flag on Windows:
<code>Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/tyler/Documents/manim/manim.py", line 5, in &ltSEMICOLONmodule&gtSEMICOLON
    manimlib.main()
  File "/home/tyler/Documents/manim/manimlib/__init__.py", line 11, in main
    scenes = manimlib.extract_scene.main(config)
  File "/home/tyler/Documents/manim/manimlib/extract_scene.py", line 104, in main
    scenes = get_scenes_to_render(all_scene_classes, config)
  File "/home/tyler/Documents/manim/manimlib/extract_scene.py", line 72, in get_scenes_to_render
    scene = scene_class(**scene_kwargs)
  File "/home/tyler/Documents/manim/manimlib/scene/scene.py", line 44, in __init__
    self.window = Window(self, **self.window_config)
  File "/home/tyler/Documents/manim/manimlib/window.py", line 21, in __init__
    super().__init__(**kwargs)
  File "/home/tyler/.local/lib/python3.7/site-packages/moderngl_window/context/pyglet/window.py", line 54, in __init__
    config=config,
  File "/home/tyler/.local/lib/python3.7/site-packages/pyglet/window/xlib/__init__.py", line 171, in __init__
    super(XlibWindow, self).__init__(*args, **kwargs)
  File "/home/tyler/.local/lib/python3.7/site-packages/pyglet/window/__init__.py", line 591, in __init__
    config = screen.get_best_config(config)
  File "/home/tyler/.local/lib/python3.7/site-packages/pyglet/canvas/base.py", line 194, in get_best_config
    raise window.NoSuchConfigException()
pyglet.window.NoSuchConfigException</code></p>
<p>Tried again with the <code>w</code> flag, but then got the following error:
```
Media will be written to ./media/. You can change this behavior with the --media_dir flag.
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "<strong>main</strong>", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/tyler/Documents/manim/manim.py", line 5, in <module>
    manimlib.main()
  File "/home/tyler/Documents/manim/manimlib/<strong>init</strong>.py", line 14, in main
    scene.run()
  File "/home/tyler/Documents/manim/manimlib/scene/scene.py", line 74, in run
    self.construct()
  File "example_scenes.py", line 84, in construct
    self.play(Transform(square, circle))
  File "/home/tyler/Documents/manim/manimlib/scene/scene.py", line 357, in wrapper
    func(self, <em>args, </em><em>kwargs)
  File "/home/tyler/Documents/manim/manimlib/scene/scene.py", line 425, in play
    self.progress_through_animations(animations)
  File "/home/tyler/Documents/manim/manimlib/scene/scene.py", line 402, in progress_through_animations
    self.update_frame(dt)
  File "/home/tyler/Documents/manim/manimlib/scene/scene.py", line 140, in update_frame
    self.camera.capture(</em>self.mobjects)
  File "/home/tyler/Documents/manim/manimlib/camera/camera.py", line 228, in capture
    shader = self.get_shader(info_group[0])
  File "/home/tyler/Documents/manim/manimlib/camera/camera.py", line 253, in get_shader
    fragment_shader=get_shader_code_from_file(info["frag"]),
  File "/home/tyler/.local/lib/python3.7/site-packages/moderngl/context.py", line 1037, in program
    varyings
moderngl.error.Error: GLSL Compiler failed</p>
<h1>fragment_shader</h1>
<p>0:179(24): error: <code>return' with wrong type int, in function</code>sdf' returning float
``</p>;1585528748000;1585528748000
605798041;588606969;qo4on;<p>You can't get hardware acceleration with virtual displays such as xvfb. Can you render <code>SquareToCircle</code> in <code>shaders</code> version correctly?</p>;1585547883000;1585547883000
605830554;588606969;timalive;<p>I was able to do it on mac - not in docker </p>;1585553299000;1585553299000
605846659;588606969;qo4on;<p>Can you try to run it in Colab? I had no success there.</p>;1585555532000;1585555532000
605947824;588606969;einarf;<p>Lengthy issue about this in the modengl project: https://github.com/moderngl/moderngl/issues/392
We do actually get it running in Google Colab with EGL, but it gets stuck in an infinite loop. That I think is a flow issue in the Manim code itself.. or at least it needs to be debugged properly to make a conclusion.</p>
<p>Manim is using moderngl and moderngl-window. I think the main issues is that it extends the pyglet window backend directly. If headless rendering is needed it should use the <code>headless.Window</code> instead.</p>
<p>The <code>BaseWindow.init_mgl_context</code> should be probably also be overridden to create the correct context type or I can add some features for this in the moderngl-window project if we figure out what is needed. Right now it tries to <strong>detect</strong> the context pyglet creates**, but in headless we just need to make our own context with no window (X11 or EGL).</p>
<p>It's not that complicated to fix thankfully.</p>
<p>I am willing to help out giving pointers for the moderngl specific things, but I don't really know much about Manim (at least yet).</p>;1585568204000;1585569436000
606115199;588606969;t-wolfeadam;<blockquote>
<p>You can't get hardware acceleration with virtual displays such as xvfb. Can you render <code>SquareToCircle</code> in <code>shaders</code> version correctly?</p>
</blockquote>
<p>Good to know! I can in Windows, but not in the Ubuntu subsystem. For both the <code>q</code> and <code>w</code> flags I get the <code>pyglet.canvas.xlib.NoSuchDisplayException: Cannot connect to "None"</code> exception.</p>;1585587005000;1585587005000
606136266;588606969;einarf;<p>I released a new version of moderngl-window what should at least solve the <code>NoSuchConfig</code> for a lot of people. You maye also have to go into the <code>window.py</code> module and set <code>samples = 0</code>.</p>
<p>@t-wolfeadam The shader error you are getting is an error in the glsl code here : https://github.com/3b1b/manim/blob/2ce0b72c440fadbfba990bdac8bc9c2ea8bff4e7/manimlib/shaders/quadratic_bezier_fill_frag.glsl#L41</p>
<p>It should return a <code>float</code>, not an <code>int</code>. Not all GLSL compilers are nice and convert constants for you automatically. I would fix that line and test it again. Make a PR as well while you are at it (if it works) 😄 </p>
<p>You may see more of those, but they are quick and easy to fix if you search for the variable or function names reported by the GLSL compiler.</p>;1585589418000;1585589554000
606164014;588606969;t-wolfeadam;<blockquote>
<p>I released a new version of moderngl-window what should at least solve the <code>NoSuchConfig</code> for a lot of people. You may also have to go into the <code>window.py</code> module and set <code>samples = 0</code>.
</p>
</blockquote>
<p>Fantastic, thanks! I can confirm for Windows that updating + setting samples to 0 fixed it!</p>
<blockquote>
<p>@t-wolfeadam The shader error you are getting is an error in the glsl code here :</p>
<p>https://github.com/3b1b/manim/blob/2ce0b72c440fadbfba990bdac8bc9c2ea8bff4e7/manimlib/shaders/quadratic_bezier_fill_frag.glsl#L41</p>
<p>It should return a <code>float</code>, not an <code>int</code>. Not all GLSL compilers are nice and convert constants for you automatically. I would fix that line and test it again. Make a PR as well while you are at it (if it works) 😄</p>
</blockquote>
<p>Ahh, so that's what sdf was referring to. That also did the trick, so thank you for that as well! (and fortunately, it was the only instance).</p>
<p>Edit: looks like someone already made the PR for that issue</p>;1585592705000;1585592791000
606171934;588606969;einarf;<p>@t-wolfeadam Yep. We just sorted out a PR after discussing the issue on the manim discord server 👍 </p>;1585593620000;1585593620000
606172498;588606969;kolibril13;<p>And here it is:
https://github.com/3b1b/manim/pull/950</p>;1585593681000;1585593681000
588343035;588343035;girishgargcool;<p>Many people face problems while installing manim. This is partly due to the fact that is isn't well documented. I myself had come across so many problems setting up manim. </p>
<p>So I have added a .ova file which will help to install manim in a virtual Ubuntu os on the host os. All the requirements and dependencies are pre-installed and the user can directly start making beautiful animations.</p>
<p>Credits: This .ova file was made by <a href="https://github.com/purusharths">Purusharth Saxena</a></p>;1585221453000;1585221453000
587571726;587571726;Enempie;<p>Hello everyone,
I'm pretty new to Python and neither had to set up different environments nor install a package (or something for it) outside of Anaconda or without the simple conda install <package> command.
As stated, I use Anaconda 3 and usually work with Spyder 4 on Windows 10 Pro 64 Bit. 
I somehow successfully installed the manim package and set up a manim environment by partially following a YT video and the instructions on this git. 
I have the following problems:
1. I can't setup a terminal in Spyder with my manim env. I guess Spyder can't find my virtualenv?
2. I can choose the manim env in VSCode and successfully render the example_scenes but only if I go to my manim directory from the terminal first which is okay but I don't know if that's the only way or if I set up something wrong (I'm not used to be forced to do that, is there a way to import manim and call the methods like I'm used to independent of the directory of my terminal)? I'm just a simple engineer and I honestly have no clue how complex software works in the background, I only know how to run my math stuff..As I said it's currently working (just from VSCode) but I'm wondering if it is possible to get it to my usual workflow (running the manim env in Spyder and especially calling manim methods without going to the directory with cd .. commands first).
I'd appreciate a helping hand a lot and thanks a lot to this amazing community in advance.</p>
<p>Sincerely </p>;1585129861000;1585154891000
603954587;587571726;Enempie;<p>Okay, 
I managed to pin it down a bit. I got Anaconda to recognize my manim env and got it to the Spyder terminal. What's now still bothering me is, that I don't seem to be able to change directory in the IPython console of Spyder and therefore can't run the extraction. Again, it works perfectly in VSCode but I'd love to have the Spyder variable explorer for my purposes.
I'll post the console log asap. </p>;1585154891000;1585154891000
585766275;585766275;Steff-san;<p>I tried to install manin for linux but i get this error:
Building wheels for collected packages: pycairo
  Building wheel for pycairo (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3.7 -u -c 'import sys, setuptools, tokenizeSEMICOLON sys.argv[0] = '"'"'/tmp/pip-install-95eiqbht/pycairo/setup.py'"'"'SEMICOLON <strong>file</strong>='"'"'/tmp/pip-install-95eiqbht/pycairo/setup.py'"'"'SEMICOLONf=getattr(tokenize, '"'"'open'"'"', open)(<strong>file</strong>)SEMICOLONcode=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"')SEMICOLONf.close()SEMICOLONexec(compile(code, <strong>file</strong>, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-vgeh7_q5
       cwd: /tmp/pip-install-95eiqbht/pycairo/
  Complete output (17 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/cairo
  copying cairo/<strong>init</strong>.py -&gtSEMICOLON build/lib.linux-x86_64-3.7/cairo
  running build_ext
  building 'cairo._cairo' extension
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/cairo
  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DPYCAIRO_VERSION_MAJOR=1 -DPYCAIRO_VERSION_MINOR=16 -DPYCAIRO_VERSION_MICRO=1 -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/python3.7m -c cairo/device.c -o build/temp.linux-x86_64-3.7/cairo/device.o
  cairo/device.c:30:10: fatal error: Python.h: No existe el archivo o el directorio
   #include <Python.h>
            ^~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1</p>
<hr />
<p>ERROR: Failed building wheel for pycairo
  Running setup.py clean for pycairo
Failed to build pycairo
Installing collected packages: pycairo
  Attempting uninstall: pycairo
    Found existing installation: pycairo 1.16.2
ERROR: Cannot uninstall 'pycairo'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.</p>
<p>Note: 
In windous i have got a similar errror.</p>;1584896980000;1585257497000
603367609;585766275;Aathish04;<p>I think you're new to Github.
It's considered bad practice to have an uninformative title and issue.
We need a bit more info from you. What OS are you using, and what were you trying to do?</p>;1585068985000;1585068985000
604357553;585766275;Nikhil0504;<blockquote>
<p>I think you're new to Github.
It's considered bad practice to have an uninformative title and issue.
We need a bit more info from you. What OS are you using, and what were you trying to do?</p>
</blockquote>
<p>he is using linux and was trying download the repo requirements </p>;1585219508000;1585219508000
198259639;198259639;Jspsun;<p>Traceback (most recent call last):
  File "<strong>main</strong>.py", line 17, in <module>
    main()
  File "<strong>main</strong>.py", line 12, in main
    text = VoiceInput.getInput()
  File "/Users/Jspsun/AtomWorkspace/Athena/Source/VoiceInputHandler.py", line 49, in getInput
    return self.getText(audio)
  File "/Users/Jspsun/AtomWorkspace/Athena/Source/VoiceInputHandler.py", line 38, in getText
    text = self.r.recognize_google(audio)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/speech_recognition/<strong>init</strong>.py", line 738, in recognize_google
    if len(actual_result.get("alternative", [])) == 0: raise UnknownValueError()
AttributeError: 'list' object has no attribute 'get'</p>
<p>To Reproduce:
- Just make a horse sound. I'm not even kidding its the easiest way to reproduce the bug</p>;1483256595000;1483256595000
318514926;318514926;Jli0423;<p>Webpack dependency not installed with 'npm install'</p>;1524855493000;1524855493000
